<html>

<head>
	<title>Human leukemia database Documentation</title>
	
	<meta name="keywords"			content="">
	<meta name="description"		content="">
	<meta http-equiv="Contect-Type"	content="text/html; charset-UTF-8">
	
	<style type="text/css">
		body
		{
			background:		#ffffff;
			color:			#000000;
			font-family:	arial, sans-serif;
			font-size:		10pt;
			margin-left: 50px;
			margin-right: 50px;
		}
		p {
			max-width: 1000px;
			text-align: left;
		}
		h1{
			margin-top 20px;
		}
		h3{
			text-decoration: underline;
			margin-bottom: 0px;
			padding-bottom: 0px;
		}
		td
		{
			font-family:	arial, sans-serif;
			font-size:		10pt;
			vertical-align: top;
		}
		.tablehead
		{
			background:		#888888;
			color:			#ffffff;
			text-align:		left;
			font-family:	arial, sans-serif;
			font-size:		10pt;
			font-weight:	bold;
		}
		th
		{
			font-family:	arial, sans-serif;
			font-size:		10pt;
			font-weight:	bold;
		}
	</style>
</head>



<body>
<h1><a name="_top_of_page">Human leukemia database</a> documentation.</h1>




<h2>Table of contents</h2>
<table width="100%"><tr>
<td>
<b><a href="#org.molgenis.auth_package">org.molgenis.auth</a></b> package:
<ul>
<li><a href="#MolgenisRole">MolgenisRole</a></li>
<li><a href="#MolgenisGroup">MolgenisGroup</a></li>
<li><a href="#MolgenisRoleGroupLink">MolgenisRoleGroupLink</a></li>
<li><a href="#Contact">Contact</a></li>
<li><a href="#Person">Person</a></li>
<li><a href="#Institute">Institute</a></li>
<li><a href="#MolgenisUser">MolgenisUser</a></li>
<li><a href="#MolgenisPermission">MolgenisPermission</a></li>
<li><a href="#Authorizable">Authorizable</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.core_package">org.molgenis.core</a></b> package:
<ul>
<li><a href="#Identifiable">Identifiable</a></li>
<li><a href="#Nameable">Nameable</a></li>
<li><a href="#Describable">Describable</a></li>
<li><a href="#OntologyTerm">OntologyTerm</a></li>
<li><a href="#Ontology">Ontology</a></li>
<li><a href="#MolgenisFile">MolgenisFile</a></li>
<li><a href="#RuntimeProperty">RuntimeProperty</a></li>
<li><a href="#Publication">Publication</a></li>
<li><a href="#UseCase">UseCase</a></li>
<li><a href="#MolgenisEntity">MolgenisEntity</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.data_package">org.molgenis.data</a></b> package:
<ul>
<li><a href="#ObservedInference">ObservedInference</a></li>
<li><a href="#DataFile">DataFile</a></li>
<li><a href="#Data">Data</a></li>
<li><a href="#DataElement">DataElement</a></li>
<li><a href="#BinaryDataMatrix">BinaryDataMatrix</a></li>
<li><a href="#CSVDataMatrix">CSVDataMatrix</a></li>
<li><a href="#DecimalDataElement">DecimalDataElement</a></li>
<li><a href="#TextDataElement">TextDataElement</a></li>
<li><a href="#OriginalFile">OriginalFile</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.organization_package">org.molgenis.organization</a></b> package:
<ul>
<li><a href="#Investigation">Investigation</a></li>
<li><a href="#InvestigationElement">InvestigationElement</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.pheno_package">org.molgenis.pheno</a></b> package:
<ul>
<li><a href="#Species">Species</a></li>
<li><a href="#AlternateId">AlternateId</a></li>
<li><a href="#ObservationElement">ObservationElement</a></li>
<li><a href="#ObservationTarget">ObservationTarget</a></li>
<li><a href="#ObservableFeature">ObservableFeature</a></li>
<li><a href="#Measurement">Measurement</a></li>
<li><a href="#Category">Category</a></li>
<li><a href="#Individual">Individual</a></li>
<li><a href="#Location">Location</a></li>
<li><a href="#Panel">Panel</a></li>
<li><a href="#Observation">Observation</a></li>
<li><a href="#ObservedValue">ObservedValue</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.protocol_package">org.molgenis.protocol</a></b> package:
<ul>
<li><a href="#Protocol">Protocol</a></li>
<li><a href="#ProtocolApplication">ProtocolApplication</a></li>
<li><a href="#ProtocolDocument">ProtocolDocument</a></li>
<li><a href="#Workflow">Workflow</a></li>
<li><a href="#WorkflowElement">WorkflowElement</a></li>
<li><a href="#WorkflowElementParameter">WorkflowElementParameter</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.xgap_package">org.molgenis.xgap</a></b> package:
<ul>
<li><a href="#Locus">Locus</a></li>
<li><a href="#Chromosome">Chromosome</a></li>
<li><a href="#NMRBin">NMRBin</a></li>
<li><a href="#Clone">Clone</a></li>
<li><a href="#DerivedTrait">DerivedTrait</a></li>
<li><a href="#EnvironmentalFactor">EnvironmentalFactor</a></li>
<li><a href="#Gene">Gene</a></li>
<li><a href="#Transcript">Transcript</a></li>
<li><a href="#Protein">Protein</a></li>
<li><a href="#Metabolite">Metabolite</a></li>
<li><a href="#Marker">Marker</a></li>
<li><a href="#SNP">SNP</a></li>
<li><a href="#Polymorphism">Polymorphism</a></li>
<li><a href="#Probe">Probe</a></li>
<li><a href="#Spot">Spot</a></li>
<li><a href="#ProbeSet">ProbeSet</a></li>
<li><a href="#MassPeak">MassPeak</a></li>
<li><a href="#InvestigationFile">InvestigationFile</a></li>
<li><a href="#Tissue">Tissue</a></li>
<li><a href="#SampleLabel">SampleLabel</a></li>
<li><a href="#Sample">Sample</a></li>
<li><a href="#PairedSample">PairedSample</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.cluster_package">org.molgenis.cluster</a></b> package:
<ul>
<li><a href="#Job">Job</a></li>
<li><a href="#Subjob">Subjob</a></li>
<li><a href="#Analysis">Analysis</a></li>
<li><a href="#ParameterSet">ParameterSet</a></li>
<li><a href="#ParameterName">ParameterName</a></li>
<li><a href="#ParameterValue">ParameterValue</a></li>
<li><a href="#DataSet">DataSet</a></li>
<li><a href="#DataName">DataName</a></li>
<li><a href="#DataValue">DataValue</a></li>
<li><a href="#SelectedParameter">SelectedParameter</a></li>
<li><a href="#SelectedData">SelectedData</a></li>
<li><a href="#RScript">RScript</a></li>
</ul>
</td>
<td>
<b><a href="#org.molgenis.hemodb_package">org.molgenis.hemodb</a></b> package:
<ul>
<li><a href="#HemoSample">HemoSample</a></li>
<li><a href="#HemoGene">HemoGene</a></li>
<li><a href="#HemoProbe">HemoProbe</a></li>
</ul>
</td>
<td>
<b><a href="#hemodb_package">All entities in hemodb</a></b>:
<ul>
<li><a href="#MolgenisRole">MolgenisRole</a></li>
<li><a href="#MolgenisGroup">MolgenisGroup</a></li>
<li><a href="#MolgenisRoleGroupLink">MolgenisRoleGroupLink</a></li>
<li><a href="#Contact">Contact</a></li>
<li><a href="#Person">Person</a></li>
<li><a href="#Institute">Institute</a></li>
<li><a href="#MolgenisUser">MolgenisUser</a></li>
<li><a href="#MolgenisPermission">MolgenisPermission</a></li>
<li><a href="#Authorizable">Authorizable</a></li>
<li><a href="#Identifiable">Identifiable</a></li>
<li><a href="#Nameable">Nameable</a></li>
<li><a href="#Describable">Describable</a></li>
<li><a href="#OntologyTerm">OntologyTerm</a></li>
<li><a href="#Ontology">Ontology</a></li>
<li><a href="#MolgenisFile">MolgenisFile</a></li>
<li><a href="#RuntimeProperty">RuntimeProperty</a></li>
<li><a href="#Publication">Publication</a></li>
<li><a href="#UseCase">UseCase</a></li>
<li><a href="#MolgenisEntity">MolgenisEntity</a></li>
<li><a href="#ObservedInference">ObservedInference</a></li>
<li><a href="#DataFile">DataFile</a></li>
<li><a href="#Data">Data</a></li>
<li><a href="#DataElement">DataElement</a></li>
<li><a href="#BinaryDataMatrix">BinaryDataMatrix</a></li>
<li><a href="#CSVDataMatrix">CSVDataMatrix</a></li>
<li><a href="#DecimalDataElement">DecimalDataElement</a></li>
<li><a href="#TextDataElement">TextDataElement</a></li>
<li><a href="#OriginalFile">OriginalFile</a></li>
<li><a href="#Investigation">Investigation</a></li>
<li><a href="#InvestigationElement">InvestigationElement</a></li>
<li><a href="#Species">Species</a></li>
<li><a href="#AlternateId">AlternateId</a></li>
<li><a href="#ObservationElement">ObservationElement</a></li>
<li><a href="#ObservationTarget">ObservationTarget</a></li>
<li><a href="#ObservableFeature">ObservableFeature</a></li>
<li><a href="#Measurement">Measurement</a></li>
<li><a href="#Category">Category</a></li>
<li><a href="#Individual">Individual</a></li>
<li><a href="#Location">Location</a></li>
<li><a href="#Panel">Panel</a></li>
<li><a href="#Observation">Observation</a></li>
<li><a href="#ObservedValue">ObservedValue</a></li>
<li><a href="#Protocol">Protocol</a></li>
<li><a href="#ProtocolApplication">ProtocolApplication</a></li>
<li><a href="#ProtocolDocument">ProtocolDocument</a></li>
<li><a href="#Workflow">Workflow</a></li>
<li><a href="#WorkflowElement">WorkflowElement</a></li>
<li><a href="#WorkflowElementParameter">WorkflowElementParameter</a></li>
<li><a href="#Locus">Locus</a></li>
<li><a href="#Chromosome">Chromosome</a></li>
<li><a href="#NMRBin">NMRBin</a></li>
<li><a href="#Clone">Clone</a></li>
<li><a href="#DerivedTrait">DerivedTrait</a></li>
<li><a href="#EnvironmentalFactor">EnvironmentalFactor</a></li>
<li><a href="#Gene">Gene</a></li>
<li><a href="#Transcript">Transcript</a></li>
<li><a href="#Protein">Protein</a></li>
<li><a href="#Metabolite">Metabolite</a></li>
<li><a href="#Marker">Marker</a></li>
<li><a href="#SNP">SNP</a></li>
<li><a href="#Polymorphism">Polymorphism</a></li>
<li><a href="#Probe">Probe</a></li>
<li><a href="#Spot">Spot</a></li>
<li><a href="#ProbeSet">ProbeSet</a></li>
<li><a href="#MassPeak">MassPeak</a></li>
<li><a href="#InvestigationFile">InvestigationFile</a></li>
<li><a href="#Tissue">Tissue</a></li>
<li><a href="#SampleLabel">SampleLabel</a></li>
<li><a href="#Sample">Sample</a></li>
<li><a href="#PairedSample">PairedSample</a></li>
<li><a href="#Job">Job</a></li>
<li><a href="#Subjob">Subjob</a></li>
<li><a href="#Analysis">Analysis</a></li>
<li><a href="#ParameterSet">ParameterSet</a></li>
<li><a href="#ParameterName">ParameterName</a></li>
<li><a href="#ParameterValue">ParameterValue</a></li>
<li><a href="#DataSet">DataSet</a></li>
<li><a href="#DataName">DataName</a></li>
<li><a href="#DataValue">DataValue</a></li>
<li><a href="#SelectedParameter">SelectedParameter</a></li>
<li><a href="#SelectedData">SelectedData</a></li>
<li><a href="#RScript">RScript</a></li>
<li><a href="#HemoSample">HemoSample</a></li>
<li><a href="#HemoGene">HemoGene</a></li>
<li><a href="#HemoProbe">HemoProbe</a></li>
</ul>
</td>
</tr></table>
<br>
<a href="#__figure_of_complete_schema">Supplementary figure: complete data model</a>


<h1><a name="org.molgenis.auth_package">org.molgenis.auth package</a></h1>

<img src="objectmodel-uml-diagram-org.molgenis.auth.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.auth.dot.png" target="_blank">show fullscreen</a>
<h3><a name="MolgenisRole">MolgenisRole</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisRole.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an MolgenisRole.
</td></tr>
</table>
<h3><a name="MolgenisGroup">MolgenisGroup</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisRole</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisGroup.
</td></tr>
</table>
<h3><a name="MolgenisRoleGroupLink">MolgenisRoleGroupLink</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>group_</b>: 
MolgenisGroup (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>role_</b>: 
MolgenisRole (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisRoleGroupLink.
</td></tr>
</table>
<h3><a name="Contact">Contact</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
 A contact is either a person or an organization. Copied
				from FuGE::Contact.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Address</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The address of the Contact.
</td></tr>
<tr>
<td 1>
<b>Phone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The telephone number of the Contact including the suitable area codes.
</td></tr>
<tr>
<td 1>
<b>Email</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The email address of the Contact.
</td></tr>
<tr>
<td 1>
<b>Fax</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The fax number of the Contact.
</td></tr>
<tr>
<td 1>
<b>tollFreePhone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A toll free phone number for the Contact, including suitable area codes.
</td></tr>
<tr>
<td 1>
<b>City</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. City of this contact.
</td></tr>
<tr>
<td 1>
<b>Country</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Country of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Contact.
</td></tr>
</table>
<h3><a name="Person">Person</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisRole</i><br>
<i> implements Contact</i><br>
</p>
<p>	


				Person represents one or more people involved with an Investigation.
				This may include authors on a paper, lab personnel or PIs. Person
				has last name, firstname, mid initial, address, contact and email. A
				Person role is included to represent how a Person is involved with
				an investigation. For submission to repository purposes an allowed
				value is 'submitter' and the term is present in the MGED Ontology,
				an alternative use could represent job title. An Example from
				ArrayExpress is E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">
					ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt.
				</a>
				.
				<br/>
				The FUGE equivalent to Person is FuGE::Person.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Address, 
Phone, 
Email, 
Fax, 
tollFreePhone, 
City, 
Country, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>FirstName</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">First Name
</td></tr>
<tr>
<td 1>
<b>MidInitials</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Mid Initials
</td></tr>
<tr>
<td 1>
<b>LastName</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Last Name
</td></tr>
<tr>
<td 1>
<b>Title</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">An academic title, e.g. Prof.dr, PhD
</td></tr>
<tr>
<td 1>
<b>Department</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Department of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Address</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Phone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The telephone number of the Contact including the suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Email</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The email address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Fax</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The fax number of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>tollFreePhone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A toll free phone number for the Contact, including suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>City</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. City of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Country</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Country of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Affiliation</b>: 
Institute (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Roles</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Indicate role of the contact, e.g. lab worker or PI. Changed from mref to xref in oct 2011.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Person.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FirstName, MidInitials, LastName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FirstName, MidInitials, LastName is unique within an Person.
</td></tr>
</table>
<h3><a name="Institute">Institute</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Contact</i><br>
</p>
<p>	
 A contact is either a person or an organization. Copied
				from FuGE::Contact.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Address, 
Phone, 
Email, 
Fax, 
tollFreePhone, 
City, 
Country, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Address</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Phone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The telephone number of the Contact including the suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Email</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The email address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Fax</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The fax number of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>tollFreePhone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A toll free phone number for the Contact, including suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>City</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. City of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Country</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Country of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Institute.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an Institute.
</td></tr>
</table>
<h3><a name="MolgenisUser">MolgenisUser</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Person</i><br>

</p>
<p>	
Anyone who can login

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Address, 
Phone, 
Email, 
Fax, 
tollFreePhone, 
City, 
Country, 
FirstName, 
MidInitials, 
LastName, 
Title, 
Affiliation, 
Department, 
Roles, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>password_</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">big fixme: password type
</td></tr>
<tr>
<td 1>
<b>activationCode</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Used as alternative authentication mechanism to verify user email and/or if user has lost password.
</td></tr>
<tr>
<td 1>
<b>active</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Boolean to indicate if this account can be used to login
</td></tr>
<tr>
<td 1>
<b>superuser</b>: bool 
(required)
</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisUser.
</td></tr>
</table>
<h3><a name="MolgenisPermission">MolgenisPermission</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>permission</b>: enum 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>role_</b>: 
MolgenisRole (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>entity</b>: 
MolgenisEntity (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisPermission.
</td></tr>
</table>
<h3><a name="Authorizable">Authorizable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
Interface for row level security

</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>canRead</b>: 
MolgenisRole (0..1)

</td>
</tr>

<tr>
<td style="color:black;">
<b>canWrite</b>: 
MolgenisRole (0..1)

</td>
</tr>

<tr>
<td style="color:black;">
<b>owns</b>: 
MolgenisRole (1..1)

</td>
</tr>

</table>
<br>
<br>
<h1><a name="org.molgenis.core_package">org.molgenis.core package</a></h1>
<p>Generic entities you can use as the starting point of
			your model.
</p>
<img src="objectmodel-uml-diagram-org.molgenis.core.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.core.dot.png" target="_blank">show fullscreen</a>
<h3><a name="Identifiable">Identifiable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
This interface assigns an automatic 'id' field to every
				object.

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Identifiable.
</td></tr>
</table>
<h3><a name="Nameable">Nameable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
This interface extends Identifiable with a unique 'name'
				field. The name length can be stored in one byte. (Java has signed
				integers) The name is by itself not unique, this should be enforced
				within a certain context. For example, unique(name, study) to make
				the name unique within one single study. The NameableDecorator
				ensures that names don't lead to problems in R and other script
				environments.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Nameable.
</td></tr>
</table>
<h3><a name="Describable">Describable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
This interface adds a description field to all entities
				the implement it.

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<h3><a name="OntologyTerm">OntologyTerm</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

				OntologyTerm defines a single entry (term) from an ontology or a
				controlled vocabulary (defined by Ontology). The name is the
				ontology term which is unique within an ontology source, such as
				[examples here]. Other data entities can reference to this
				OntologyTerm to harmonize naming of concepts. Each term should have
				a local, unique label within the Investigation. If no suitable
				ontology term exists then one can define new terms locally (in which
				case there is no formal accession for the term limiting its use for
				cross-Investigation queries). In those cases the local name should
				be repeated in both term and termAccession. Maps to
				FuGE::OntologyIndividual; in MAGE-TAB there is no separate entity to
				model terms.
				<br/>
				Optionally a local controlled vocabulary or ontology can be defined,
				for example to represent 'Codelists' often used in questionaires.
				Note: this is not a InvestigationElement because of the additional
				xref_label and unique constraint.This class defines a single entry
				from an ontology or a controlled vocabulary.
				<br/>
				If it is a simple controlled vocabulary, there may be no formal
				accession for the term. In these cases the local name should be
				repeated in both term and termAccession. If the term has a value,
				the OntologyTerm will have a single DataProperty whose value was the
				value for the property. For instance, for an OntologyIndividual
				based on the MO ontology the attributes might be: The term would be
				what is usually called the local name in the Ontology, for instance
				'Age'; The termAccession could be
				'http://mged.sourceforge.net/ontologies/MGEDOntology.owl#Age' or a
				an arbitrary accession if one exists; The identifier is a unique
				identifier for individuals in the scope of the FuGE instance; The
				inherited name attribute should not be used; The ontologyURI of
				OntologySource could be
				'http://mged.sourceforge.net/ontologies/MGEDOntology.owl". The
				OntologyTerm subclasses are instances of Ontology classes and
				properties, not the actual terms themselves. An OntologyIndividual,
				if based on an existing Ontology, can be considered a statement that
				can be validated against the referenced ontology. The subclasses and
				their associations are based on the Ontology Definition Model,
				ad/2005-04-13, submitted to the OMG as a response to RFP
				ad/2003-03-40, Copyright 2005 DSTC Pty Ltd. Copyright 2005 IBM
				Copyright 2005 Sandpiper Software, Inc under the standard OMG
				license terms. 
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>termAccession</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.
</td></tr>
<tr>
<td 1>
<b>definition</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) The definition of the term.
</td></tr>
<tr>
<td 1>
<b>termPath</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">EXTENSION. The Ontology Lookup Service path that contains this term.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ontology</b>: 
Ontology (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an OntologyTerm.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(ontology, termAccession)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The term accession is unique within an Ontology.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(ontology, name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
ADDITION: Name is unique within its type .
</td></tr>
</table>
<h3><a name="Ontology">Ontology</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	
 Ontology defines a reference to a an ontology or
				controlled vocabulary from which well-defined and stable (ontology)
				terms can be obtained. Each Ontology should have a unique name, for
				instance: Gene Ontology, Mammalian Phenotype, Human Phenotype
				Ontology, Unified Medical Language System, Medical Subject Headings,
				etc. Also a abbreviation is required, for instance: GO, MP, HPO,
				UMLS, MeSH, etc. Use of existing ontologies/vocabularies is
				recommended to harmonize phenotypic feature and value descriptions.
				But one can also create a 'local' Ontology. The Ontology class maps
				to FuGE::Ontology, MAGE-TAB::TermSourceREF.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>ontologyAccession</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A identifier that uniquely identifies the ontology (typically an acronym). E.g. GO, MeSH, HPO.
</td></tr>
<tr>
<td 1>
<b>ontologyURI</b>: hyperlink 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) A URI that references the location of the ontology.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Ontology.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within its type .
</td></tr>
</table>
<h3><a name="MolgenisFile">MolgenisFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

				Helper entity to deal with files. Has a decorator to regulate
				storage and coupling to an Entity. Do not make abstract because of
				subtyping. This means the names of the subclasses will be used to
				distinguish MolgenisFiles and place them in the correct folders.
				<br/>
				MS: make it use the &lt;field type="file" property under the hood. 
				<br/>
				MS: where do the mimetypes go? I mean, I don't see the added value
				now.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Extension</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The file extension. This will be mapped to MIME type at runtime. For example, a type 'png' will be served out as 'image/png'.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisFile.
</td></tr>
</table>
<h3><a name="RuntimeProperty">RuntimeProperty</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an RuntimeProperty.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within its type .
</td></tr>
</table>
<h3><a name="Publication">Publication</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	


				Publication is part of the Investigation package and is used to
				represent information about one or more publications related to an
				Investigation. The publication need not only be primary publication
				for an Investigation but may also represent other related
				information- though this use is less common. Publications have
				attributes of publications Authors and also DOI and Pubmed
				identifiers (when these are available). These are represented as
				OntologyTerms as in the MAGE-TAB model all 'xrefs' (cross
				references) for ontologies and accession numbers are handled
				generically. An example of a publication is available in an IDF file
				from ArrayExpress is experiment E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt
				</a>
				.
				<br/>
				The FuGE equivalent to Publication is FuGE::Bibliographic Reference.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>authorList</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The names of the authors of the publication
</td></tr>
<tr>
<td 1>
<b>Title</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The title of the Publication
</td></tr>
<tr>
<td 1>
<b>Year</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The year of the Publication
</td></tr>
<tr>
<td 1>
<b>Journal</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The title of the Journal
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>PubmedID</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Pubmed ID</td></tr>
<tr>
<td style="color:#333333;">
<b>DOI</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Publication DOI</td></tr>
<tr>
<td style="color:#333333;">
<b>Status</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The status of the Publication</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Publication.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an Publication.
</td></tr>
</table>
<h3><a name="UseCase">UseCase</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
 All the use cases send to the server are stored in this
				entity .

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>UseCaseId</b>: int 
(required)
</td>
</tr>

<tr>
<td 1>
<b>UseCaseName</b>: string 
(required)
</td>
</tr>

<tr>
<td 1>
<b>SearchType</b>: string 
(required)
</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(UseCaseId)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field UseCaseId is unique within an UseCase.
</td></tr>
</table>
<h3><a name="MolgenisEntity">MolgenisEntity</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
Referenceable catalog of entity names, menus, forms and
				plugins.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Name of the entity
</td></tr>
<tr>
<td 1>
<b>type_</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Type of the entity
</td></tr>
<tr>
<td 1>
<b>className</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Full name of the entity
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisEntity.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(className)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field className is unique within an MolgenisEntity.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, type_)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields name, type_ is unique within an MolgenisEntity.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.data_package">org.molgenis.data package</a></h1>
<p>This package enables data models to treat part of their
			data model as data matrix. is essential for Entity Attribute Value
			modeling such as used in xgap and pheno.
</p>
<img src="objectmodel-uml-diagram-org.molgenis.data.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.data.dot.png" target="_blank">show fullscreen</a>
<h3><a name="ObservedInference">ObservedInference</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservedValue</i><br>

</p>
<p>	
 The InferredValue class defines ObservedValues that are
				inferred as result of human or computational post-processing of
				other ObservedValues. For example: hypertensive = yes when mean
				arterial pressure = 135 AND no hypertension affecting medicine is
				taken. The optional inferenceType property enables definition of the
				type of inference. For example 'mean of a lenght': the observable
				feature is 'length' and the inferenceType is 'mean'. Just as with
				regular ObservedValues, the protocol used for this inference can be
				defined via the protocolApplication association that is inherited
				from ObservedValue. The InferredValue class has no direct mapping to
				other models: XGAP would use input/ouput Data; PaGE would use a self
				reference on ObservedValue.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
ontologyReference, 
value, 
relation, 
time, 
endtime, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>derivedFrom</b>: ObservedValue (1..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">References to one or more observed values that were used to infer this observation. TODO: needs multicolumn lookup</td></tr>
<tr>
<td style="color:black;">
<b>inferenceType</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the ontology definition for this inference. For example: mean</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservedInference.
</td></tr>
</table>
<h3><a name="DataFile">DataFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
ObservedFile is to store observations that result in a
				file. Mapping to other models: MAGE-TAB 1.1 has the column
				ArrayDataFile and DerivedArrayDataFile. In order to make the
				MAGE-TAB 1.1. model more generic we have generalized these to
				DataFile and provided named associations to the respective types via
				Scan and Assay. TODO: make this link to MolgenisFile? Or distinguish
				between links and data?

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>uri</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">reference to the location of the file
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>format</b>: 
OntologyTerm (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">format of the file. Discussion: is this not already solved in MolgenisFile</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataFile.
</td></tr>
</table>
<h3><a name="Data">Data</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ProtocolApplication</i><br>

</p>
<p>	

				Data is a data structure to store a homogeneous matrix of
				observedvalues as one unit, that is, all data elements in the set
				have the same type of feature, target and value. For example: an
				expression qtlProfile (observation.feature) for a Panel of mouse
				(observation.target) that consists of a matrix of Probe X marker
				(featureType and targetType respectively). In the user interface we
				expect that this observation can be shown as a bigger set of
				observations but click-able so the user can drill down to the
				underlying matrix.
				<br/>
				Data is also an observationTarget: this allows Data to be referred
				to in an ObservedValue.relation. TODO: describe how this can be used
				to define inputs/outputs for a protocolApplication. This would allow
				us to use it to link 'pheno' to 'cluster' package so that the whole
				provenace can be administrated as part of the observation models.
				<br/>
				This class maps to XGAP.DataMatrix and MAGE-TAB.Data.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
time, 
protocol, 
Performer, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>FeatureType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Defines the type of the columns of this data set. Each column refers to a Feature or Subject.
<br>enum_options: 'Chromosome', 'DerivedTrait', 'EnvironmentalFactor', 'Gene', 'Individual', 'Marker', 'MassPeak', 'Measurement', 'Metabolite', 'Panel', 'Probe', 'ProbeSet', 'Sample', 'Spot', 'SNP', 'Promoter', 'ChipPeak'
</td></tr>
<tr>
<td 1>
<b>TargetType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Defines the type of the rows of this matrix. Each row refers to a Feature or Subject.
<br>enum_options: 'Chromosome', 'DerivedTrait', 'EnvironmentalFactor', 'Gene', 'Individual', 'Marker', 'MassPeak', 'Measurement', 'Metabolite', 'Panel', 'Probe', 'ProbeSet', 'Sample', 'Spot', 'SNP', 'Promoter', 'ChipPeak'
</td></tr>
<tr>
<td 1>
<b>ValueType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Type of the values of this matrix, either text strings or decimal numbers.
<br>enum_options: 'Text', 'Decimal'
</td></tr>
<tr>
<td 1>
<b>Storage</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Tells you how the data elements are stored or should be stored. For example, 'Binary'.
<br>enum_options: 'Database', 'Binary', 'CSV'
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Data.
</td></tr>
<tr>
<td 1>
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
name is unique within data matrix.
</td></tr>
</table>
<h3><a name="DataElement">DataElement</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Observation</i><br>
</p>
<p>	
 Extension of observed value that keeps track of the
				position in a Data matrix and can be used to store the elements of
				Data. Copied from XGAP. MAGE-TAB 1.1. has the column ArrayDataMatrix
				and DerivedArrayDataMatrix. The formats of both types are
				tab-delimited and the distinction is that an ArrayDataMatrix is raw
				data and the DerivedArrayDataMatrix is processed data. The
				DerivedArrayDataMatrix associates a file of a specific format -
				seehe MAGE-TAB specification for details. In order to index the Rows
				and Columns of DataMatrix we have generated a DimensionElement that
				allows us to index DimensionRow and DimensionColumn.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>FeatureIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row position in the matrix.
</td></tr>
<tr>
<td 1>
<b>TargetIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Col position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the data set this entity belongs to.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FeatureIndex, TargetIndex, Data)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FeatureIndex, TargetIndex, Data is unique within an DataElement.
</td></tr>
</table>
<h3><a name="BinaryDataMatrix">BinaryDataMatrix</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	
Binary file backend for a datamatrix. This extension is
				used to deal with the actual source file. Coupled to a matrix with
				source type 'BinaryFile'. This entity is not shown in the interface.
				Discussion: I am not so happy with the need of alternative
				subclasses. Instead you just need a driver.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the datamatrix this binary file belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an BinaryDataMatrix.
</td></tr>
</table>
<h3><a name="CSVDataMatrix">CSVDataMatrix</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	
CSV file backend for a datamatrix. Convenient to deal
				with the actual source file. Coupled to a matrix with source type
				'CSVFile'. This entity is not shown in the interface.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the datamatrix this CSV file belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an CSVDataMatrix.
</td></tr>
</table>
<h3><a name="DecimalDataElement">DecimalDataElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements DataElement</i><br>
</p>
<p>	
 A DataElement for storing decimal data.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
Data, 
FeatureIndex, 
TargetIndex, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The value, e.g., correlation.
</td></tr>
<tr>
<td style="color:#333333;">
<b>FeatureIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>TargetIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Col position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the data set this entity belongs to.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DecimalDataElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FeatureIndex, TargetIndex, Data)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FeatureIndex, TargetIndex, Data is unique within an DecimalDataElement.
</td></tr>
</table>
<h3><a name="TextDataElement">TextDataElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements DataElement</i><br>
</p>
<p>	
Store text data

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
Data, 
FeatureIndex, 
TargetIndex, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The value, e.g., genotype strings like AA, BA, BB.
</td></tr>
<tr>
<td style="color:#333333;">
<b>FeatureIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>TargetIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Col position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the data set this entity belongs to.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an TextDataElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FeatureIndex, TargetIndex, Data)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FeatureIndex, TargetIndex, Data is unique within an TextDataElement.
</td></tr>
</table>
<h3><a name="OriginalFile">OriginalFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	
An unmodified original file that belongs to this
				datamatrix.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the datamatrix this file belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an OriginalFile.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.organization_package">org.molgenis.organization package</a></h1>
<p>Generic entities you can use as the starting point of
			your model.
</p>
<img src="objectmodel-uml-diagram-org.molgenis.organization.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.organization.dot.png" target="_blank">show fullscreen</a>
<h3><a name="Investigation">Investigation</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable, Describable</i><br>
</p>
<p>	
Investigation defines self-contained units of study. For
				example: Framingham study. Optionally a description and an accession
				to a data source can be provided. Each Investigation has a unique
				name and a group of subjects of observation (ObservableTarget),
				traits of observation (ObservableFeature), results (in
				ObservedValues), and optionally actions (Protocols,
				ProtoclApplications). 'Invetigation' maps to standard XGAP/FuGE
				Investigation, MAGE-TAB Experiment and METABASE:Study.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>startDate</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The start point of the study.
</td></tr>
<tr>
<td 1>
<b>endDate</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The end point of the study.
</td></tr>
<tr>
<td 1>
<b>accession</b>: hyperlink 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) URI or accession number to indicate source of Study. E.g. arrayexpress:M-EXP-2345
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>contacts</b>: Person (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Contact persons for this study</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Investigation.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="InvestigationElement">InvestigationElement</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable, Describable</i><br>
</p>
<p>	
General superclass to use for all elements of an
				Investigation. The unique forces all of these elements to have
				unique names within one Investigation. (can be changed to
				Name,Study,Type by flipping a boolean in the molgenis 'Unique'
				class.)

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an InvestigationElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.pheno_package">org.molgenis.pheno package</a></h1>
<p>Pheno is an generic version of XGAP developed in close
			collaboration within GEN2PHEN, EBI, UMC Groningen, U Groningen, FIMM,
			U Leicester. Todo: add docs again from pheno model. The pheno core
			needs to be preserved! Add changelog special section.
		
</p>
<img src="objectmodel-uml-diagram-org.molgenis.pheno.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.pheno.dot.png" target="_blank">show fullscreen</a>
<h3><a name="Species">Species</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	
Ontology terms for species. E.g. Arabidopsis thaliana.
				DISCUSSION: should we avoid subclasses of OntologyTerm and instead
				make a 'tag' filter on terms so we can make pulldowns context
				dependent (e.g. to only show particular subqueries of ontologies).
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Species.
</td></tr>
</table>
<h3><a name="AlternateId">AlternateId</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	

				An external identifier for an annotation. For example: name='R13H8.1', ontology='ensembl' or name='WBgene00000912', ontology='wormbase'.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an AlternateId.
</td></tr>
</table>
<h3><a name="ObservationElement">ObservationElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements InvestigationElement</i><br>
</p>
<p>	
 Elements that are the targets or features we are
				looking at of our research.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>label</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">User friendly textual representation of this ObservationElement. For example: 'male',     'mouse 3 in cage 7' or 'TRA-2 like protein'. Label allows for human-readable name that is potentially not unique.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>AlternateId</b>: AlternateId (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Alternative identifiers or symbols that this element is known by.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservationElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an ObservationElement.
</td></tr>
</table>
<h3><a name="ObservationTarget">ObservationTarget</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationElement</i><br>

</p>
<p>	
 An ObservationTarget class defines the subjects of
				observation. For instance: individual 1 from Investigation x. The
				ObservationTarget class maps to XGAP:Subject, METABASE:Patient and
				maps to Page:Abstract_Observation_Target. The name of
				observationTargets is unique.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservationTarget.
</td></tr>
</table>
<h3><a name="ObservableFeature">ObservableFeature</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationElement</i><br>

</p>
<p>	
ObservableFeature defines anything that can be observed
				in a phenotypic Investigation. For instance: Height, Systolic blood
				pressure, Diastolic blood pressure, and Treatment for hypertension
				are observable features. The name of ObservableFeature is unique
				within one Investigation. It is recommended that each
				ObservableFeature is named according to a well-defined ontology term
				which can be specified via ontologyReference. Note that in some
				instances an observableFeature can also be an observationTarget, for
				example in the case of correlation matrices. The ObservableFeature
				class maps to XGAP:Trait, METABASE:Question, FuGE:DimensionElement,
				and PaGE:ObservableFeature. Multi-value features can be grouped by
				Protocol. For instance: high blood pressure can be inferred from
				observations for features systolic and diastolic blood pressure.
				There may be many alternative protocols to measure a feature. See
				Protocol section.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservableFeature.
</td></tr>
</table>
<h3><a name="Measurement">Measurement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
Generic obserable feature to flexibly define a
				measurement
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>dataType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the technical data type. E.g. 'int'
<br>enum_options: 'xref', 'string', 'categorical', 'date', 'datetime', 'int', 'code', 'image', 'decimal', 'bool', 'file', 'log', 'data', 'exe'
</td></tr>
<tr>
<td 1>
<b>temporal</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false)
</td></tr>
<tr>
<td 1>
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended
</td></tr>
<tr>
<td 1>
<b>panelLabelAllowedForRelation</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Label that must have been applied to the Panel that      can be linked to (through the 'relation' field in ObservedValue) when using this       Measurement (example: a Measurement 'Species' can only result      in ObservedValues that have relations to Panels labeled as 'Species')
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>unit</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg</td></tr>
<tr>
<td 1>
<b>categories</b>: Category (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Translation of codes into categories if applicable</td></tr>
<tr>
<td 1>
<b>targettypeAllowedForRelation</b>: 
MolgenisEntity (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Subclass of ObservationTarget (Individual, Panel or Location) that      can be linked to (through the 'relation' field in ObservedValue) when using this       Measurement (example: a Measurement 'Species' can only result in ObservedValues       that have relations to Panels)</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Measurement.
</td></tr>
</table>
<h3><a name="Category">Category</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationElement</i><br>

</p>
<p>	
Special kind of ObservationElement to define categorical answer codes such as are
				often used in Questionaires. A list of categories can be attached to an
				Measurement using Measurement.categories. For example the
				Measurement 'sex' has {code_string = 1, label=male} and {code_string = 2,
				label=female}. Categories can be linked to well-defined ontology terms
				via the ontologyReference. Category extends ObservationElement such that
				it can be referenced by ObservedValue.value. The Category class maps
				to METABASE::Category
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>code_string</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The code used to represent this category. For example: { '1' codes for 'male', '2'-'female'}
</td></tr>
<tr>
<td 1>
<b>description</b>: text 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Description of the code.     Use of ontology terms references to establish unambigious descriptions is     recommended
</td></tr>
<tr>
<td 1>
<b>isMissing</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">whether this code should be treated as missing value.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this code.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Category.
</td></tr>
</table>
<h3><a name="Individual">Individual</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
 The Individuals class defines human cases that are used
				as observation target. The Individual class maps to XGAP:Individual
				and PaGE:Individual. Note that minimal information like 'sex' can be
				defined as ObservedValue, and that that basic relationships like
				'father' and 'mother' can also be defined via ObservedRelationship,
				using the 'relation' field. Groups of individuals can be defined via
				Panel.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Mother</b>: 
Individual (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Refers to the mother of the individual.</td></tr>
<tr>
<td style="color:black;">
<b>Father</b>: 
Individual (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Refers to the father of the individual.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Individual.
</td></tr>
</table>
<h3><a name="Location">Location</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
This class defines physical locations such as buildings,
				departments, rooms, freezers and cages. Use ObservedValues to link
				locations to eachother, to build a location hierarchy.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Location.
</td></tr>
</table>
<h3><a name="Panel">Panel</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
The Panel class defines groups of individuals based on
				cohort design, case/controls, families, etc. For instance: LifeLines
				cohort, 'middle aged man', 'recombinant mouse inbred Line dba x b6'
				or 'Smith family'. A Panel can act as a single ObservationTarget.
				For example: average height (ObservedValue) in the LifeLines cohort
				(Panel) is 174cm. The Panel class maps to XGAP:Strain and PaGE:Panel
				classes. In METABASE this is assumed there is one panel per study.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Individuals</b>: Individual (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The list of individuals in this panel</td></tr>
<tr>
<td style="color:black;">
<b>Species</b>: 
Species (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The species this panel is an instance of/part of/extracted from.</td></tr>
<tr>
<td style="color:black;">
<b>PanelType</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Indicate the type of Panel (example: Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution)</td></tr>
<tr>
<td style="color:black;">
<b>FounderPanels</b>: Panel (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The panel(s) that were used to create this panel.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Panel.
</td></tr>
</table>
<h3><a name="Observation">Observation</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
FIXME: change to ObservationTarget?

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Observation.
</td></tr>
</table>
<h3><a name="ObservedValue">ObservedValue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable, Observation</i><br>
</p>
<p>	

				Generic storage of values, relationships and optional ontology
				mapping of the value/relation. Values can be atomatic observations,
				e.g., length (feature) of individual 1 (target) = 179cm (value).
				Values can also be relationship values, e.g., extract (feature) of
				sample 1 (target) = derived sample (relation).
				<br/>
				Discussion: how to model sample pooling in this model?
				<br/>
				More Discussion: do we want to have type specific subclasses? No,
				because you can solve this by casting during querying?
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>value</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The value observed
</td></tr>
<tr>
<td 1>
<b>time</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Time when the value was observed. For example in time series or if feature is time-dependent like 'age'
</td></tr>
<tr>
<td 1>
<b>endtime</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Time when the value's validity ended
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the     ontology definition or 'code' for this value (recommended for non-numeric     values such as codes)</td></tr>
<tr>
<td style="color:#333333;">
<b>relation</b>: 
ObservationElement (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to other end of the relationship, if any. For example to a 'brother' or from 'sample' to 'derivedSample'.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservedValue.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.protocol_package">org.molgenis.protocol package</a></h1>
<p>Molgenis compute framework that extends the molgenis
			protocol framework adding the computational details
		
</p>
<img src="objectmodel-uml-diagram-org.molgenis.protocol.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.protocol.dot.png" target="_blank">show fullscreen</a>
<h3><a name="Protocol">Protocol</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements InvestigationElement</i><br>
</p>
<p>	


				The Protocol class defines parameterizable descriptions of methods;
				each protocol has a unique name within an Study. Each
				ProtocolApplication can define the ObservableFeatures it can
				observe. Also the protocol parameters can be modeled using
				ObservableFeatures (Users are expected to 'tag' the observeable
				feature by setting ObserveableFeature type as 'ProtocolParameter'.
				Examples of protocols are: SOP for blood pressure measurement used
				by UK biobank, or 'R/qtl' as protocol for statistical analysis.
				Protocol is a high level object that represents the details of
				protocols used during the investigation. The uses of Protocols to
				process BioMaterials and Data are referenced by ProtocolApplication
				(in the SDRF part of the format). Protocol has an association to
				OntologyTerm to represent the type of protocol. Protocols are
				associated with Hardware, Software and Parameters used in the
				Protocol. An example from ArrayExpress is E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">
					ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt.
				</a>
    <br/>
				The FUGE equivalent to Protocol is FuGE::Protocol.
				<br/>
				The Protocol class maps to FuGE/XGAP/MageTab Protocol, but in
				contrast to FuGE it is not required to extend protocol before use.
				The Protocol class also maps to METABASE:Form (note that components
				are solved during METABASE:Visit which can be nested). Has no
				equivalent in PaGE.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>description</b>: richtext 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Description, or reference to a description, of the protocol
</td></tr>
<tr>
<td 1>
<b>TargetFilter</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Expression that filters the InvestigationElements that can be targetted using this protocol.     This helps the user to only select from targets that matter when setting observedvalues. For example: type='individual' AND species = 'human'
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ProtocolType</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">annotation of the protocol to a well-defined ontological class.</td></tr>
<tr>
<td style="color:#333333;">
<b>Features</b>: ObservableFeature (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The features that can be observed using this protocol. For example 'length' or 'rs123534' or 'probe123'. Also protocol parameters are considered observable features as they are important to the interpretation of the observed values.</td></tr>
<tr>
<td style="color:#333333;">
<b>Contact</b>: 
Person (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">TODO Check if there can be multiple contacts.</td></tr>
<tr>
<td style="color:#333333;">
<b>subprotocols</b>: Protocol (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Subprotocols of this protocol</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Protocol.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
</table>
<h3><a name="ProtocolApplication">ProtocolApplication</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements InvestigationElement</i><br>
</p>
<p>	

				A ProtocolApplication class defines the actual action of observation
				by refering to a protocol and optional ParameterValues. The name
				field can be used to label applications with a human understandeable
				tag. For example: the action of blood pressure measurement on 1000
				individuals, using a particular protocol, resulting in 1000
				associated observed values. If desired, protocols can be shared
				between Studys; in those cases one should simply refer to a protocol
				in another Study.
				<br/>
				ProtocolApplications are used in MAGE-TAB format to reference to
				protocols used, with optionally use of certain protocol parameter
				values. For example, a Source may be transformed into a Labeled
				Extract by the subsequent application of a Extraction and Labeling
				protocol. ProtocolApplication is associated with and Edge that links
				input/output, e.g. Source to Labeled Extract. The order of the
				application of protocols can be set in order to be able to
				reconstruct the left-to-right order of protocol references in
				MAGE-TAB format. The FuGE equivalent to ProtocolApplication is
				FuGE:ProtocolApplication, however input/output is modeled using
				Edge.
				<br/>
				The ProtocolApplication class maps to FuGE/XGAP ProtocolApplication,
				but in FuGE ProtocolApplications can take Material or Data (or both)
				as input and produce Material or Data (or both) as output. Similar
				to PaGE.ObservationMethod. Maps to METABASE:Visit (also note that
				METABASE:PlannedVisit allows for planning of protocol applications;
				this is outside scope for this model?).
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>time</b>: datetime 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">time when the protocol was applied.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>protocol</b>: 
Protocol (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol that is being used.</td></tr>
<tr>
<td style="color:#333333;">
<b>Performer</b>: Person (0..n)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ProtocolApplication.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
</table>
<h3><a name="ProtocolDocument">ProtocolDocument</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>document</b>: file 
(required)
</td>
</tr>

</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>protocol</b>: 
Protocol (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ProtocolDocument.
</td></tr>
</table>
<h3><a name="Workflow">Workflow</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Protocol</i><br>
<i> implements Identifiable, Nameable</i><br>
</p>
<p>	
A workflow is a plan to execute a series of subprotocols
				in a particular order. Each workflow elements is another protocol as
				refered to via WorkflowElement. Because Workflow extends Protocol,
				workflows can be nested just as any other protocol.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
ProtocolType, 
Features, 
TargetFilter, 
Contact, 
subprotocols, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Workflow.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an Workflow.
</td></tr>
</table>
<h3><a name="WorkflowElement">WorkflowElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	
Elements of a workflow are references to protocols. The
				whole workflow is a directed graph with each element pointing to the
				previousSteps that the current workflow element depends on.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Workflow</b>: 
Workflow (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Workflow this element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>protocol</b>: 
Protocol (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Protocol to be used at this workflow step</td></tr>
<tr>
<td style="color:#333333;">
<b>PreviousSteps</b>: WorkflowElement (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Previous steps that need to be done before this protocol can be executed.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an WorkflowElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an WorkflowElement.
</td></tr>
</table>
<h3><a name="WorkflowElementParameter">WorkflowElementParameter</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
Element parameters are the way to link workflow elements
				together. It allows override of the parameters from the previous
				step.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Value of this parameter. Can be a template of form ${other} refering to previous values in context.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>WorkflowElement</b>: 
WorkflowElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">To attach a parameter to a WorkflowElement</td></tr>
<tr>
<td style="color:#333333;">
<b>Parameter</b>: 
ObservableFeature (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Parameter definition.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an WorkflowElementParameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(WorkflowElement, Parameter)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Feature value is set only once per element.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.xgap_package">org.molgenis.xgap package</a></h1>
<p>XGAP
</p>
<img src="objectmodel-uml-diagram-org.molgenis.xgap.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.xgap.dot.png" target="_blank">show fullscreen</a>
<h3><a name="Locus">Locus</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
 position. Typical examples of such traits are genes,
				probes and markers. Common structure for entities that have a
				genomic

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td 1>
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td 1>
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td 1>
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td 1>
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<h3><a name="Chromosome">Chromosome</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>orderNr</b>: int 
(required)
</td>
</tr>

<tr>
<td 1>
<b>isAutosomal</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Is 'yes' when number of chromosomes is equal in male and female individuals, i.e., if not a sex chromosome.
</td></tr>
<tr>
<td 1>
<b>bpLength</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Lenght of the chromsome in base pairs.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Species</b>: 
Species (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the species this     chromosome belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Chromosome.
</td></tr>
</table>
<h3><a name="NMRBin">NMRBin</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Shift of the NMR frequency due to the chemical
				environment.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an NMRBin.
</td></tr>
</table>
<h3><a name="Clone">Clone</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 BAC clone fragment.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Clone.
</td></tr>
</table>
<h3><a name="DerivedTrait">DerivedTrait</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Any meta trait, eg. false discovery rates, P-values,
				thresholds.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DerivedTrait.
</td></tr>
</table>
<h3><a name="EnvironmentalFactor">EnvironmentalFactor</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
Experimental conditions,
			such as temperature differences, batch effects etc.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an EnvironmentalFactor.
</td></tr>
</table>
<h3><a name="Gene">Gene</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 Trait annotations specific for genes.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Main symbol this gene is known by (not necessarily unique, in constrast to 'name')
</td></tr>
<tr>
<td 1>
<b>Orientation</b>: enum 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Orientation of the gene on the genome (F=forward, R=reverse)
<br>enum_options: 'F', 'R'
</td></tr>
<tr>
<td 1>
<b>Control</b>: bool 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Indicating whether this is a 'housekeeping' gene that can be used as control.
</td></tr>
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Gene.
</td></tr>
</table>
<h3><a name="Transcript">Transcript</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for transcripts.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Gene</b>: 
Gene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene that produces this protein</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Transcript.
</td></tr>
</table>
<h3><a name="Protein">Protein</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for proteins.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>AminoSequence</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The aminoacid sequence.
</td></tr>
<tr>
<td 1>
<b>Mass</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The mass of this metabolite
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Gene</b>: 
Gene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene that produces this protein</td></tr>
<tr>
<td 1>
<b>Transcript</b>: 
Transcript (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The transcript variant that produces this protein</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Protein.
</td></tr>
</table>
<h3><a name="Metabolite">Metabolite</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for metabolites.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Formula</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The chemical formula of a metabolite.
</td></tr>
<tr>
<td 1>
<b>Mass</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The mass of this metabolite
</td></tr>
<tr>
<td 1>
<b>Structure</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The chemical structure of a metabolite (in SMILES representation).
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Metabolite.
</td></tr>
</table>
<h3><a name="Marker">Marker</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 Trait annotations specific for markers.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ReportsFor</b>: Marker (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The marker (or a subclass like 'SNP') this marker (or a subclass like 'SNP') reports for.</td></tr>
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Marker.
</td></tr>
</table>
<h3><a name="SNP">SNP</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Marker</i><br>

</p>
<p>	
A SNP is a special kind of Marker, but can also be seen as a phenotype to map against in some cases. A single-nucleotide polymorphism is a DNA sequence variation occurring when a single nucleotide in the genome (or other shared sequence) differs between members of a biological species or paired chromosomes in an individual.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
ReportsFor, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Status</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The status of this SNP, eg 'confirmed'.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Polymorphism</b>: Polymorphism (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The polymorphism that belongs to this SNP.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SNP.
</td></tr>
</table>
<h3><a name="Polymorphism">Polymorphism</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
The difference of a single base discovered between two sequenced individuals.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Base</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The affected DNA base. Note that you can select the reference base here.
<br>enum_options: 'Reference', 'A', 'T', 'C', 'G'
</td></tr>
<tr>
<td 1>
<b>Value</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The strain/genotype for which this polymorphism was discovered. E.g. 'N2' or 'CB4856'.
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Polymorphism.
</td></tr>
</table>
<h3><a name="Probe">Probe</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 A piece of sequence that reports for the expression of
				a gene, typically spotted onto a microarray.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Mismatch</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Indicating whether the probe is a match
</td></tr>
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ProbeSet</b>: 
ProbeSet (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Optional: probeset this probe belongs to (e.g., in Affymetrix assays).</td></tr>
<tr>
<td style="color:#333333;">
<b>ReportsFor</b>: 
Gene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene this probe reports for.</td></tr>
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Probe.
</td></tr>
</table>
<h3><a name="Spot">Spot</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Probe</i><br>

</p>
<p>	

				This is the spot on a microarray.
				<br/>
				Note: We don't distinquish between probes (the sequence) and spots
				(the sequence as spotted on the array).
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
Mismatch, 
ProbeSet, 
ReportsFor, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>x</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row
</td></tr>
<tr>
<td 1>
<b>y</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Column
</td></tr>
<tr>
<td 1>
<b>GridX</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Meta Row
</td></tr>
<tr>
<td 1>
<b>GridY</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Meta Column
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Spot.
</td></tr>
<tr>
<td 1>
<b>unique(x, y, GridX, GridY)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields x, y, GridX, GridY is unique within an Spot.
</td></tr>
</table>
<h3><a name="ProbeSet">ProbeSet</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 A set of Probes. E.g. an Affymetrix probeset has multiple
				probes. It implements locus because sometimes you want to give the complete set of probes a range,
				for example: indicating that this set of probes spans basepair 0 through 10.000.000 on chromosome 3.
				The same information could arguably also be queried from the probes themselves, but if you have 40k probes,
				retrieving the same information from only ProbeSet (if annotated so) would be much faster.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ProbeSet.
</td></tr>
</table>
<h3><a name="MassPeak">MassPeak</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 A peak that has been selected within a mass
				spectrometry experiment.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>MZ</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Mass over charge ratio of this peak.
</td></tr>
<tr>
<td 1>
<b>RetentionTime</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The retention-time of this peak in minutes.
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MassPeak.
</td></tr>
</table>
<h3><a name="InvestigationFile">InvestigationFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Investigation</b>: 
Investigation (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an InvestigationFile.
</td></tr>
</table>
<h3><a name="Tissue">Tissue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Tissue.
</td></tr>
</table>
<h3><a name="SampleLabel">SampleLabel</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SampleLabel.
</td></tr>
</table>
<h3><a name="Sample">Sample</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	


</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Individual</b>: 
Individual (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The individual from which this sample was taken.</td></tr>
<tr>
<td style="color:black;">
<b>Tissue</b>: 
Tissue (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The tissue from which this sample was taken.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Sample.
</td></tr>
</table>
<h3><a name="PairedSample">PairedSample</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
 A pair of samples labeled for a two-color microarray
				experiment.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Subject1</b>: 
Individual (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The first subject</td></tr>
<tr>
<td style="color:black;">
<b>Label1</b>: 
SampleLabel (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Which channel or Fluorescent labeling is associated with the first subject</td></tr>
<tr>
<td style="color:black;">
<b>Subject2</b>: 
Individual (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The second sample</td></tr>
<tr>
<td style="color:black;">
<b>Label2</b>: 
SampleLabel (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Which channel or Fluorescent labeling is associated with the second subject</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an PairedSample.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.cluster_package">org.molgenis.cluster package</a></h1>
<p>Cluster calculation tables.
</p>
<img src="objectmodel-uml-diagram-org.molgenis.cluster.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.cluster.dot.png" target="_blank">show fullscreen</a>
<h3><a name="Job">Job</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>OutputDataName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Name of the matrix that will be written
</td></tr>
<tr>
<td 1>
<b>Timestamp</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Datatime when the job was started.
</td></tr>
<tr>
<td 1>
<b>ComputeResource</b>: enum 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Analysis</b>: 
Analysis (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Job.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(OutputDataName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
OutputDataName is unique.
</td></tr>
</table>
<h3><a name="Subjob">Subjob</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>StatusCode</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Status code of this subjob.
</td></tr>
<tr>
<td 1>
<b>StatusText</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Status text of this subjob.
</td></tr>
<tr>
<td 1>
<b>StatusProgress</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Percentage done.
</td></tr>
<tr>
<td 1>
<b>Nr</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Number of this subjob within the job.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Job</b>: 
Job (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the job this subjob belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Subjob.
</td></tr>
</table>
<h3><a name="Analysis">Analysis</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Optional description of this type of analysis.
</td></tr>
<tr>
<td 1>
<b>TargetFunctionName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The function used to start a specific type of analysis on the cluster.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ParameterSet</b>: 
ParameterSet (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>DataSet</b>: 
DataSet (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Analysis.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="ParameterSet">ParameterSet</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ParameterSet.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="ParameterName">ParameterName</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Optional description of this parameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ParameterSet</b>: 
ParameterSet (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ParameterName.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, ParameterSet)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a ParameterSet.
</td></tr>
</table>
<h3><a name="ParameterValue">ParameterValue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Possible value of this parameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ParameterName</b>: 
ParameterName (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ParameterValue.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, ParameterName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a ParameterName.
</td></tr>
</table>
<h3><a name="DataSet">DataSet</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataSet.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="DataName">DataName</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>DataSet</b>: 
DataSet (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataName.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, DataSet)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a dataset.
</td></tr>
</table>
<h3><a name="DataValue">DataValue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>DataName</b>: 
DataName (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Value</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Possible reference of this Data.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataValue.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, DataName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a DataName.
</td></tr>
</table>
<h3><a name="SelectedParameter">SelectedParameter</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>ParameterName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied name of this parameter.
</td></tr>
<tr>
<td 1>
<b>ParameterValue</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied value of this parameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Job</b>: 
Job (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SelectedParameter.
</td></tr>
</table>
<h3><a name="SelectedData">SelectedData</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>DataName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied name of this Data
</td></tr>
<tr>
<td 1>
<b>DataValue</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied referenced name of this Data
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Job</b>: 
Job (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SelectedData.
</td></tr>
</table>
<h3><a name="RScript">RScript</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends InvestigationFile</i><br>

</p>
<p>	

			Proof of concept to show users can add scripts to database, to be replaced later with more generic version from compute model.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
Extension, 
Investigation, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an RScript.
</td></tr>
</table>
<br>
<br>
<h1><a name="org.molgenis.hemodb_package">org.molgenis.hemodb package</a></h1>

<img src="objectmodel-uml-diagram-org.molgenis.hemodb.dot.png" style="border: solid thin black;">
<a href="objectmodel-uml-diagram-org.molgenis.hemodb.dot.png" target="_blank">show fullscreen</a>
<h3><a name="HemoSample">HemoSample</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	


</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Illumina internal ID
</td></tr>
<tr>
<td 1>
<b>Platform</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Illumina Platform
</td></tr>
<tr>
<td 1>
<b>RNA_Amplification</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Method of amplification (Magnetic or Column)
</td></tr>
<tr>
<td 1>
<b>Sample</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">UMCG sample ID
</td></tr>
<tr>
<td 1>
<b>expr_CD34</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Expression of CD34, plus = expressed, minus = not expressed
</td></tr>
<tr>
<td 1>
<b>AML_normal</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Sample type
</td></tr>
<tr>
<td 1>
<b>Sample_Group</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Sample classification (ALL CD34+, AML CD34-, AML CD34+, AML MNC, BC CML, BM CD34+, BM post ASCT, BM retroviral model, CB CD34+, CB retroviral model, mob PBSC, NA, NSG, stroma)
</td></tr>
<tr>
<td 1>
<b>age</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">NA, old, young, fetal, numerical
</td></tr>
<tr>
<td 1>
<b>FAB</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">French-American-British classification system of hematologic diseases
</td></tr>
<tr>
<td 1>
<b>risk</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Risk group (Poor, intermediate, good)
</td></tr>
<tr>
<td 1>
<b>FLT3_class</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Classification based on the FLT3 gene (ITD,NA,normal,WT)
</td></tr>
<tr>
<td 1>
<b>NPM</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Classification based on NPM (cyt,NA,not determined,wt)
</td></tr>
<tr>
<td 1>
<b>PROCENT_CD34</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Percentage of CD34
</td></tr>
<tr>
<td 1>
<b>PB_BM_cb</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Sample origin (peripheral blood, bone marrow or cord blood)
</td></tr>
<tr>
<td 1>
<b>Age_years</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Age in years
</td></tr>
<tr>
<td 1>
<b>Cytogenetics</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Cytogenetics (where NK = Normal Karyotype)
</td></tr>
<tr>
<td 1>
<b>OS_months</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Overall survival
</td></tr>
<tr>
<td 1>
<b>EFS_months</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Event free survival
</td></tr>
<tr>
<td 1>
<b>CR</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Complete remission
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an HemoSample.
</td></tr>
</table>
<h3><a name="HemoGene">HemoGene</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for genes.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Main symbol this gene is known by (not necessarily unique, in constrast to 'name')
</td></tr>
<tr>
<td 1>
<b>Control</b>: bool 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Indicating whether this is a 'housekeeping' gene that can be used as control.
</td></tr>
<tr>
<td 1>
<b>Orientation</b>: enum 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Orientation of the gene on the genome (F=forward, R=reverse)
<br>enum_options: '?', 'F', 'R'
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an HemoGene.
</td></tr>
</table>
<h3><a name="HemoProbe">HemoProbe</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 A piece of sequence that reports for the expression of
				a gene, typically spotted onto a microarray.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>EntrezID</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The entrez ID for the probe
</td></tr>
<tr>
<td 1>
<b>RefseqID</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The refseq ID for the probe
</td></tr>
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ProbeSet</b>: 
ProbeSet (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Optional: probeset this probe belongs to (e.g., in Affymetrix assays).</td></tr>
<tr>
<td style="color:#333333;">
<b>ReportsFor</b>: 
HemoGene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene this probe reports for.</td></tr>
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an HemoProbe.
</td></tr>
</table>
<br>
<br>

<h1><a name="hemodb_package">hemodb package</a></h1>
<h3><a name="MolgenisRole">MolgenisRole</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisRole.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an MolgenisRole.
</td></tr>
</table>
<h3><a name="MolgenisGroup">MolgenisGroup</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisRole</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisGroup.
</td></tr>
</table>
<h3><a name="MolgenisRoleGroupLink">MolgenisRoleGroupLink</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>group_</b>: 
MolgenisGroup (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>role_</b>: 
MolgenisRole (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisRoleGroupLink.
</td></tr>
</table>
<h3><a name="Contact">Contact</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
 A contact is either a person or an organization. Copied
				from FuGE::Contact.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Address</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The address of the Contact.
</td></tr>
<tr>
<td 1>
<b>Phone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The telephone number of the Contact including the suitable area codes.
</td></tr>
<tr>
<td 1>
<b>Email</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The email address of the Contact.
</td></tr>
<tr>
<td 1>
<b>Fax</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The fax number of the Contact.
</td></tr>
<tr>
<td 1>
<b>tollFreePhone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A toll free phone number for the Contact, including suitable area codes.
</td></tr>
<tr>
<td 1>
<b>City</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. City of this contact.
</td></tr>
<tr>
<td 1>
<b>Country</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Country of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Contact.
</td></tr>
</table>
<h3><a name="Person">Person</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisRole</i><br>
<i> implements Contact</i><br>
</p>
<p>	


				Person represents one or more people involved with an Investigation.
				This may include authors on a paper, lab personnel or PIs. Person
				has last name, firstname, mid initial, address, contact and email. A
				Person role is included to represent how a Person is involved with
				an investigation. For submission to repository purposes an allowed
				value is 'submitter' and the term is present in the MGED Ontology,
				an alternative use could represent job title. An Example from
				ArrayExpress is E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">
					ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt.
				</a>
				.
				<br/>
				The FUGE equivalent to Person is FuGE::Person.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Address, 
Phone, 
Email, 
Fax, 
tollFreePhone, 
City, 
Country, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>FirstName</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">First Name
</td></tr>
<tr>
<td 1>
<b>MidInitials</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Mid Initials
</td></tr>
<tr>
<td 1>
<b>LastName</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Last Name
</td></tr>
<tr>
<td 1>
<b>Title</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">An academic title, e.g. Prof.dr, PhD
</td></tr>
<tr>
<td 1>
<b>Department</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Department of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Address</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Phone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The telephone number of the Contact including the suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Email</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The email address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Fax</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The fax number of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>tollFreePhone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A toll free phone number for the Contact, including suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>City</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. City of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Country</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Country of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Affiliation</b>: 
Institute (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Roles</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Indicate role of the contact, e.g. lab worker or PI. Changed from mref to xref in oct 2011.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Person.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FirstName, MidInitials, LastName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FirstName, MidInitials, LastName is unique within an Person.
</td></tr>
</table>
<h3><a name="Institute">Institute</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Contact</i><br>
</p>
<p>	
 A contact is either a person or an organization. Copied
				from FuGE::Contact.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Address, 
Phone, 
Email, 
Fax, 
tollFreePhone, 
City, 
Country, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Address</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Phone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The telephone number of the Contact including the suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Email</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The email address of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Fax</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The fax number of the Contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>tollFreePhone</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A toll free phone number for the Contact, including suitable area codes.
</td></tr>
<tr>
<td style="color:#333333;">
<b>City</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. City of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Country</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Added from the old definition of MolgenisUser. Country of this contact.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Institute.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an Institute.
</td></tr>
</table>
<h3><a name="MolgenisUser">MolgenisUser</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Person</i><br>

</p>
<p>	
Anyone who can login

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Address, 
Phone, 
Email, 
Fax, 
tollFreePhone, 
City, 
Country, 
FirstName, 
MidInitials, 
LastName, 
Title, 
Affiliation, 
Department, 
Roles, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>password_</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">big fixme: password type
</td></tr>
<tr>
<td 1>
<b>activationCode</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Used as alternative authentication mechanism to verify user email and/or if user has lost password.
</td></tr>
<tr>
<td 1>
<b>active</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Boolean to indicate if this account can be used to login
</td></tr>
<tr>
<td 1>
<b>superuser</b>: bool 
(required)
</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisUser.
</td></tr>
</table>
<h3><a name="MolgenisPermission">MolgenisPermission</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>permission</b>: enum 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>role_</b>: 
MolgenisRole (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>entity</b>: 
MolgenisEntity (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisPermission.
</td></tr>
</table>
<h3><a name="Authorizable">Authorizable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
Interface for row level security

</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>canRead</b>: 
MolgenisRole (0..1)

</td>
</tr>

<tr>
<td style="color:black;">
<b>canWrite</b>: 
MolgenisRole (0..1)

</td>
</tr>

<tr>
<td style="color:black;">
<b>owns</b>: 
MolgenisRole (1..1)

</td>
</tr>

</table>
<h3><a name="Identifiable">Identifiable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
This interface assigns an automatic 'id' field to every
				object.

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Identifiable.
</td></tr>
</table>
<h3><a name="Nameable">Nameable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
This interface extends Identifiable with a unique 'name'
				field. The name length can be stored in one byte. (Java has signed
				integers) The name is by itself not unique, this should be enforced
				within a certain context. For example, unique(name, study) to make
				the name unique within one single study. The NameableDecorator
				ensures that names don't lead to problems in R and other script
				environments.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Nameable.
</td></tr>
</table>
<h3><a name="Describable">Describable</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
This interface adds a description field to all entities
				the implement it.

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<h3><a name="OntologyTerm">OntologyTerm</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

				OntologyTerm defines a single entry (term) from an ontology or a
				controlled vocabulary (defined by Ontology). The name is the
				ontology term which is unique within an ontology source, such as
				[examples here]. Other data entities can reference to this
				OntologyTerm to harmonize naming of concepts. Each term should have
				a local, unique label within the Investigation. If no suitable
				ontology term exists then one can define new terms locally (in which
				case there is no formal accession for the term limiting its use for
				cross-Investigation queries). In those cases the local name should
				be repeated in both term and termAccession. Maps to
				FuGE::OntologyIndividual; in MAGE-TAB there is no separate entity to
				model terms.
				<br/>
				Optionally a local controlled vocabulary or ontology can be defined,
				for example to represent 'Codelists' often used in questionaires.
				Note: this is not a InvestigationElement because of the additional
				xref_label and unique constraint.This class defines a single entry
				from an ontology or a controlled vocabulary.
				<br/>
				If it is a simple controlled vocabulary, there may be no formal
				accession for the term. In these cases the local name should be
				repeated in both term and termAccession. If the term has a value,
				the OntologyTerm will have a single DataProperty whose value was the
				value for the property. For instance, for an OntologyIndividual
				based on the MO ontology the attributes might be: The term would be
				what is usually called the local name in the Ontology, for instance
				'Age'; The termAccession could be
				'http://mged.sourceforge.net/ontologies/MGEDOntology.owl#Age' or a
				an arbitrary accession if one exists; The identifier is a unique
				identifier for individuals in the scope of the FuGE instance; The
				inherited name attribute should not be used; The ontologyURI of
				OntologySource could be
				'http://mged.sourceforge.net/ontologies/MGEDOntology.owl". The
				OntologyTerm subclasses are instances of Ontology classes and
				properties, not the actual terms themselves. An OntologyIndividual,
				if based on an existing Ontology, can be considered a statement that
				can be validated against the referenced ontology. The subclasses and
				their associations are based on the Ontology Definition Model,
				ad/2005-04-13, submitted to the OMG as a response to RFP
				ad/2003-03-40, Copyright 2005 DSTC Pty Ltd. Copyright 2005 IBM
				Copyright 2005 Sandpiper Software, Inc under the standard OMG
				license terms. 
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>termAccession</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) The accession number assigned to the ontology term in its source ontology. If empty it is assumed to be a locally defined term.
</td></tr>
<tr>
<td 1>
<b>definition</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) The definition of the term.
</td></tr>
<tr>
<td 1>
<b>termPath</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">EXTENSION. The Ontology Lookup Service path that contains this term.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ontology</b>: 
Ontology (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) The source ontology or controlled vocabulary list that ontology terms have been obtained from.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an OntologyTerm.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(ontology, termAccession)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The term accession is unique within an Ontology.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(ontology, name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
ADDITION: Name is unique within its type .
</td></tr>
</table>
<h3><a name="Ontology">Ontology</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	
 Ontology defines a reference to a an ontology or
				controlled vocabulary from which well-defined and stable (ontology)
				terms can be obtained. Each Ontology should have a unique name, for
				instance: Gene Ontology, Mammalian Phenotype, Human Phenotype
				Ontology, Unified Medical Language System, Medical Subject Headings,
				etc. Also a abbreviation is required, for instance: GO, MP, HPO,
				UMLS, MeSH, etc. Use of existing ontologies/vocabularies is
				recommended to harmonize phenotypic feature and value descriptions.
				But one can also create a 'local' Ontology. The Ontology class maps
				to FuGE::Ontology, MAGE-TAB::TermSourceREF.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>ontologyAccession</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">A identifier that uniquely identifies the ontology (typically an acronym). E.g. GO, MeSH, HPO.
</td></tr>
<tr>
<td 1>
<b>ontologyURI</b>: hyperlink 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) A URI that references the location of the ontology.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Ontology.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within its type .
</td></tr>
</table>
<h3><a name="MolgenisFile">MolgenisFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

				Helper entity to deal with files. Has a decorator to regulate
				storage and coupling to an Entity. Do not make abstract because of
				subtyping. This means the names of the subclasses will be used to
				distinguish MolgenisFiles and place them in the correct folders.
				<br/>
				MS: make it use the &lt;field type="file" property under the hood. 
				<br/>
				MS: where do the mimetypes go? I mean, I don't see the added value
				now.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Extension</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The file extension. This will be mapped to MIME type at runtime. For example, a type 'png' will be served out as 'image/png'.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisFile.
</td></tr>
</table>
<h3><a name="RuntimeProperty">RuntimeProperty</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an RuntimeProperty.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within its type .
</td></tr>
</table>
<h3><a name="Publication">Publication</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	


				Publication is part of the Investigation package and is used to
				represent information about one or more publications related to an
				Investigation. The publication need not only be primary publication
				for an Investigation but may also represent other related
				information- though this use is less common. Publications have
				attributes of publications Authors and also DOI and Pubmed
				identifiers (when these are available). These are represented as
				OntologyTerms as in the MAGE-TAB model all 'xrefs' (cross
				references) for ontologies and accession numbers are handled
				generically. An example of a publication is available in an IDF file
				from ArrayExpress is experiment E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt
				</a>
				.
				<br/>
				The FuGE equivalent to Publication is FuGE::Bibliographic Reference.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>authorList</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The names of the authors of the publication
</td></tr>
<tr>
<td 1>
<b>Title</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The title of the Publication
</td></tr>
<tr>
<td 1>
<b>Year</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The year of the Publication
</td></tr>
<tr>
<td 1>
<b>Journal</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The title of the Journal
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>PubmedID</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Pubmed ID</td></tr>
<tr>
<td style="color:#333333;">
<b>DOI</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Publication DOI</td></tr>
<tr>
<td style="color:#333333;">
<b>Status</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The status of the Publication</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Publication.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an Publication.
</td></tr>
</table>
<h3><a name="UseCase">UseCase</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
 All the use cases send to the server are stored in this
				entity .

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>UseCaseId</b>: int 
(required)
</td>
</tr>

<tr>
<td 1>
<b>UseCaseName</b>: string 
(required)
</td>
</tr>

<tr>
<td 1>
<b>SearchType</b>: string 
(required)
</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(UseCaseId)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field UseCaseId is unique within an UseCase.
</td></tr>
</table>
<h3><a name="MolgenisEntity">MolgenisEntity</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
Referenceable catalog of entity names, menus, forms and
				plugins.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Name of the entity
</td></tr>
<tr>
<td 1>
<b>type_</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Type of the entity
</td></tr>
<tr>
<td 1>
<b>className</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Full name of the entity
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MolgenisEntity.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(className)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field className is unique within an MolgenisEntity.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, type_)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields name, type_ is unique within an MolgenisEntity.
</td></tr>
</table>
<h3><a name="ObservedInference">ObservedInference</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservedValue</i><br>

</p>
<p>	
 The InferredValue class defines ObservedValues that are
				inferred as result of human or computational post-processing of
				other ObservedValues. For example: hypertensive = yes when mean
				arterial pressure = 135 AND no hypertension affecting medicine is
				taken. The optional inferenceType property enables definition of the
				type of inference. For example 'mean of a lenght': the observable
				feature is 'length' and the inferenceType is 'mean'. Just as with
				regular ObservedValues, the protocol used for this inference can be
				defined via the protocolApplication association that is inherited
				from ObservedValue. The InferredValue class has no direct mapping to
				other models: XGAP would use input/ouput Data; PaGE would use a self
				reference on ObservedValue.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
ontologyReference, 
value, 
relation, 
time, 
endtime, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>derivedFrom</b>: ObservedValue (1..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">References to one or more observed values that were used to infer this observation. TODO: needs multicolumn lookup</td></tr>
<tr>
<td style="color:black;">
<b>inferenceType</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the ontology definition for this inference. For example: mean</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservedInference.
</td></tr>
</table>
<h3><a name="DataFile">DataFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
ObservedFile is to store observations that result in a
				file. Mapping to other models: MAGE-TAB 1.1 has the column
				ArrayDataFile and DerivedArrayDataFile. In order to make the
				MAGE-TAB 1.1. model more generic we have generalized these to
				DataFile and provided named associations to the respective types via
				Scan and Assay. TODO: make this link to MolgenisFile? Or distinguish
				between links and data?

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>uri</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">reference to the location of the file
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>format</b>: 
OntologyTerm (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">format of the file. Discussion: is this not already solved in MolgenisFile</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataFile.
</td></tr>
</table>
<h3><a name="Data">Data</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ProtocolApplication</i><br>

</p>
<p>	

				Data is a data structure to store a homogeneous matrix of
				observedvalues as one unit, that is, all data elements in the set
				have the same type of feature, target and value. For example: an
				expression qtlProfile (observation.feature) for a Panel of mouse
				(observation.target) that consists of a matrix of Probe X marker
				(featureType and targetType respectively). In the user interface we
				expect that this observation can be shown as a bigger set of
				observations but click-able so the user can drill down to the
				underlying matrix.
				<br/>
				Data is also an observationTarget: this allows Data to be referred
				to in an ObservedValue.relation. TODO: describe how this can be used
				to define inputs/outputs for a protocolApplication. This would allow
				us to use it to link 'pheno' to 'cluster' package so that the whole
				provenace can be administrated as part of the observation models.
				<br/>
				This class maps to XGAP.DataMatrix and MAGE-TAB.Data.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
time, 
protocol, 
Performer, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>FeatureType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Defines the type of the columns of this data set. Each column refers to a Feature or Subject.
<br>enum_options: 'Chromosome', 'DerivedTrait', 'EnvironmentalFactor', 'Gene', 'Individual', 'Marker', 'MassPeak', 'Measurement', 'Metabolite', 'Panel', 'Probe', 'ProbeSet', 'Sample', 'Spot', 'SNP', 'Promoter', 'ChipPeak'
</td></tr>
<tr>
<td 1>
<b>TargetType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Defines the type of the rows of this matrix. Each row refers to a Feature or Subject.
<br>enum_options: 'Chromosome', 'DerivedTrait', 'EnvironmentalFactor', 'Gene', 'Individual', 'Marker', 'MassPeak', 'Measurement', 'Metabolite', 'Panel', 'Probe', 'ProbeSet', 'Sample', 'Spot', 'SNP', 'Promoter', 'ChipPeak'
</td></tr>
<tr>
<td 1>
<b>ValueType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Type of the values of this matrix, either text strings or decimal numbers.
<br>enum_options: 'Text', 'Decimal'
</td></tr>
<tr>
<td 1>
<b>Storage</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Tells you how the data elements are stored or should be stored. For example, 'Binary'.
<br>enum_options: 'Database', 'Binary', 'CSV'
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Data.
</td></tr>
<tr>
<td 1>
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
name is unique within data matrix.
</td></tr>
</table>
<h3><a name="DataElement">DataElement</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Observation</i><br>
</p>
<p>	
 Extension of observed value that keeps track of the
				position in a Data matrix and can be used to store the elements of
				Data. Copied from XGAP. MAGE-TAB 1.1. has the column ArrayDataMatrix
				and DerivedArrayDataMatrix. The formats of both types are
				tab-delimited and the distinction is that an ArrayDataMatrix is raw
				data and the DerivedArrayDataMatrix is processed data. The
				DerivedArrayDataMatrix associates a file of a specific format -
				seehe MAGE-TAB specification for details. In order to index the Rows
				and Columns of DataMatrix we have generated a DimensionElement that
				allows us to index DimensionRow and DimensionColumn.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>FeatureIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row position in the matrix.
</td></tr>
<tr>
<td 1>
<b>TargetIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Col position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the data set this entity belongs to.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FeatureIndex, TargetIndex, Data)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FeatureIndex, TargetIndex, Data is unique within an DataElement.
</td></tr>
</table>
<h3><a name="BinaryDataMatrix">BinaryDataMatrix</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	
Binary file backend for a datamatrix. This extension is
				used to deal with the actual source file. Coupled to a matrix with
				source type 'BinaryFile'. This entity is not shown in the interface.
				Discussion: I am not so happy with the need of alternative
				subclasses. Instead you just need a driver.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the datamatrix this binary file belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an BinaryDataMatrix.
</td></tr>
</table>
<h3><a name="CSVDataMatrix">CSVDataMatrix</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	
CSV file backend for a datamatrix. Convenient to deal
				with the actual source file. Coupled to a matrix with source type
				'CSVFile'. This entity is not shown in the interface.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the datamatrix this CSV file belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an CSVDataMatrix.
</td></tr>
</table>
<h3><a name="DecimalDataElement">DecimalDataElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements DataElement</i><br>
</p>
<p>	
 A DataElement for storing decimal data.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
Data, 
FeatureIndex, 
TargetIndex, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The value, e.g., correlation.
</td></tr>
<tr>
<td style="color:#333333;">
<b>FeatureIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>TargetIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Col position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the data set this entity belongs to.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DecimalDataElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FeatureIndex, TargetIndex, Data)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FeatureIndex, TargetIndex, Data is unique within an DecimalDataElement.
</td></tr>
</table>
<h3><a name="TextDataElement">TextDataElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements DataElement</i><br>
</p>
<p>	
Store text data

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
Data, 
FeatureIndex, 
TargetIndex, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The value, e.g., genotype strings like AA, BA, BB.
</td></tr>
<tr>
<td style="color:#333333;">
<b>FeatureIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>TargetIndex</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Col position in the matrix.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the data set this entity belongs to.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an TextDataElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(FeatureIndex, TargetIndex, Data)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields FeatureIndex, TargetIndex, Data is unique within an TextDataElement.
</td></tr>
</table>
<h3><a name="OriginalFile">OriginalFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	
An unmodified original file that belongs to this
				datamatrix.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Data</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the datamatrix this file belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an OriginalFile.
</td></tr>
</table>
<h3><a name="Investigation">Investigation</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable, Describable</i><br>
</p>
<p>	
Investigation defines self-contained units of study. For
				example: Framingham study. Optionally a description and an accession
				to a data source can be provided. Each Investigation has a unique
				name and a group of subjects of observation (ObservableTarget),
				traits of observation (ObservableFeature), results (in
				ObservedValues), and optionally actions (Protocols,
				ProtoclApplications). 'Invetigation' maps to standard XGAP/FuGE
				Investigation, MAGE-TAB Experiment and METABASE:Study.

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>startDate</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The start point of the study.
</td></tr>
<tr>
<td 1>
<b>endDate</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The end point of the study.
</td></tr>
<tr>
<td 1>
<b>accession</b>: hyperlink 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) URI or accession number to indicate source of Study. E.g. arrayexpress:M-EXP-2345
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>contacts</b>: Person (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Contact persons for this study</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Investigation.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="InvestigationElement">InvestigationElement</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable, Describable</i><br>
</p>
<p>	
General superclass to use for all elements of an
				Investigation. The unique forces all of these elements to have
				unique names within one Investigation. (can be changed to
				Name,Study,Type by flipping a boolean in the molgenis 'Unique'
				class.)

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an InvestigationElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
</table>
<h3><a name="Species">Species</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	
Ontology terms for species. E.g. Arabidopsis thaliana.
				DISCUSSION: should we avoid subclasses of OntologyTerm and instead
				make a 'tag' filter on terms so we can make pulldowns context
				dependent (e.g. to only show particular subqueries of ontologies).
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Species.
</td></tr>
</table>
<h3><a name="AlternateId">AlternateId</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	

				An external identifier for an annotation. For example: name='R13H8.1', ontology='ensembl' or name='WBgene00000912', ontology='wormbase'.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an AlternateId.
</td></tr>
</table>
<h3><a name="ObservationElement">ObservationElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements InvestigationElement</i><br>
</p>
<p>	
 Elements that are the targets or features we are
				looking at of our research.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>label</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">User friendly textual representation of this ObservationElement. For example: 'male',     'mouse 3 in cage 7' or 'TRA-2 like protein'. Label allows for human-readable name that is potentially not unique.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>AlternateId</b>: AlternateId (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Alternative identifiers or symbols that this element is known by.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservationElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an ObservationElement.
</td></tr>
</table>
<h3><a name="ObservationTarget">ObservationTarget</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationElement</i><br>

</p>
<p>	
 An ObservationTarget class defines the subjects of
				observation. For instance: individual 1 from Investigation x. The
				ObservationTarget class maps to XGAP:Subject, METABASE:Patient and
				maps to Page:Abstract_Observation_Target. The name of
				observationTargets is unique.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservationTarget.
</td></tr>
</table>
<h3><a name="ObservableFeature">ObservableFeature</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationElement</i><br>

</p>
<p>	
ObservableFeature defines anything that can be observed
				in a phenotypic Investigation. For instance: Height, Systolic blood
				pressure, Diastolic blood pressure, and Treatment for hypertension
				are observable features. The name of ObservableFeature is unique
				within one Investigation. It is recommended that each
				ObservableFeature is named according to a well-defined ontology term
				which can be specified via ontologyReference. Note that in some
				instances an observableFeature can also be an observationTarget, for
				example in the case of correlation matrices. The ObservableFeature
				class maps to XGAP:Trait, METABASE:Question, FuGE:DimensionElement,
				and PaGE:ObservableFeature. Multi-value features can be grouped by
				Protocol. For instance: high blood pressure can be inferred from
				observations for features systolic and diastolic blood pressure.
				There may be many alternative protocols to measure a feature. See
				Protocol section.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservableFeature.
</td></tr>
</table>
<h3><a name="Measurement">Measurement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
Generic obserable feature to flexibly define a
				measurement
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>dataType</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the technical data type. E.g. 'int'
<br>enum_options: 'xref', 'string', 'categorical', 'date', 'datetime', 'int', 'code', 'image', 'decimal', 'bool', 'file', 'log', 'data', 'exe'
</td></tr>
<tr>
<td 1>
<b>temporal</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Whether this feature is time dependent and can have different values when measured       on different times (e.g. weight, temporal=true) or generally only measured once (e.g. birth date,       temporal=false)
</td></tr>
<tr>
<td 1>
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Rudimentary meta data about the observable feature. Use of ontology       terms references to establish unambigious descriptions is recommended
</td></tr>
<tr>
<td 1>
<b>panelLabelAllowedForRelation</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Label that must have been applied to the Panel that      can be linked to (through the 'relation' field in ObservedValue) when using this       Measurement (example: a Measurement 'Species' can only result      in ObservedValues that have relations to Panels labeled as 'Species')
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>unit</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the well-defined measurement unit used to observe this feature       (if feature is that concrete). E.g. mmHg</td></tr>
<tr>
<td 1>
<b>categories</b>: Category (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Translation of codes into categories if applicable</td></tr>
<tr>
<td 1>
<b>targettypeAllowedForRelation</b>: 
MolgenisEntity (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Subclass of ObservationTarget (Individual, Panel or Location) that      can be linked to (through the 'relation' field in ObservedValue) when using this       Measurement (example: a Measurement 'Species' can only result in ObservedValues       that have relations to Panels)</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Measurement.
</td></tr>
</table>
<h3><a name="Category">Category</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationElement</i><br>

</p>
<p>	
Special kind of ObservationElement to define categorical answer codes such as are
				often used in Questionaires. A list of categories can be attached to an
				Measurement using Measurement.categories. For example the
				Measurement 'sex' has {code_string = 1, label=male} and {code_string = 2,
				label=female}. Categories can be linked to well-defined ontology terms
				via the ontologyReference. Category extends ObservationElement such that
				it can be referenced by ObservedValue.value. The Category class maps
				to METABASE::Category
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>code_string</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The code used to represent this category. For example: { '1' codes for 'male', '2'-'female'}
</td></tr>
<tr>
<td 1>
<b>description</b>: text 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Description of the code.     Use of ontology terms references to establish unambigious descriptions is     recommended
</td></tr>
<tr>
<td 1>
<b>isMissing</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">whether this code should be treated as missing value.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this code.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Category.
</td></tr>
</table>
<h3><a name="Individual">Individual</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
 The Individuals class defines human cases that are used
				as observation target. The Individual class maps to XGAP:Individual
				and PaGE:Individual. Note that minimal information like 'sex' can be
				defined as ObservedValue, and that that basic relationships like
				'father' and 'mother' can also be defined via ObservedRelationship,
				using the 'relation' field. Groups of individuals can be defined via
				Panel.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Mother</b>: 
Individual (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Refers to the mother of the individual.</td></tr>
<tr>
<td style="color:black;">
<b>Father</b>: 
Individual (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Refers to the father of the individual.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Individual.
</td></tr>
</table>
<h3><a name="Location">Location</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
This class defines physical locations such as buildings,
				departments, rooms, freezers and cages. Use ObservedValues to link
				locations to eachother, to build a location hierarchy.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Location.
</td></tr>
</table>
<h3><a name="Panel">Panel</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
The Panel class defines groups of individuals based on
				cohort design, case/controls, families, etc. For instance: LifeLines
				cohort, 'middle aged man', 'recombinant mouse inbred Line dba x b6'
				or 'Smith family'. A Panel can act as a single ObservationTarget.
				For example: average height (ObservedValue) in the LifeLines cohort
				(Panel) is 174cm. The Panel class maps to XGAP:Strain and PaGE:Panel
				classes. In METABASE this is assumed there is one panel per study.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Individuals</b>: Individual (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The list of individuals in this panel</td></tr>
<tr>
<td style="color:black;">
<b>Species</b>: 
Species (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The species this panel is an instance of/part of/extracted from.</td></tr>
<tr>
<td style="color:black;">
<b>PanelType</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Indicate the type of Panel (example: Natural=wild type, Parental=parents of a cross, F1=First generation of cross, RCC=Recombinant congenic, CSS=chromosome substitution)</td></tr>
<tr>
<td style="color:black;">
<b>FounderPanels</b>: Panel (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The panel(s) that were used to create this panel.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Panel.
</td></tr>
</table>
<h3><a name="Observation">Observation</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
FIXME: change to ObservationTarget?

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Observation.
</td></tr>
</table>
<h3><a name="ObservedValue">ObservedValue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable, Observation</i><br>
</p>
<p>	

				Generic storage of values, relationships and optional ontology
				mapping of the value/relation. Values can be atomatic observations,
				e.g., length (feature) of individual 1 (target) = 179cm (value).
				Values can also be relationship values, e.g., extract (feature) of
				sample 1 (target) = derived sample (relation).
				<br/>
				Discussion: how to model sample pooling in this model?
				<br/>
				More Discussion: do we want to have type specific subclasses? No,
				because you can solve this by casting during querying?
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
Investigation, 
protocolApplication, 
Feature, 
Target, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>value</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The value observed
</td></tr>
<tr>
<td 1>
<b>time</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Time when the value was observed. For example in time series or if feature is time-dependent like 'age'
</td></tr>
<tr>
<td 1>
<b>endtime</b>: datetime 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Time when the value's validity ended
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the     ontology definition or 'code' for this value (recommended for non-numeric     values such as codes)</td></tr>
<tr>
<td style="color:#333333;">
<b>relation</b>: 
ObservationElement (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to other end of the relationship, if any. For example to a 'brother' or from 'sample' to 'derivedSample'.</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>protocolApplication</b>: 
ProtocolApplication (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol application that was used to produce this observation. For example a particular patient visit or the application of a microarray or the calculation of a QTL model</td></tr>
<tr>
<td style="color:#333333;">
<b>Feature</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservableFeature that this observation was made on. For example 'probe123'. Can be ommited for 1D data (i.e., a data list).</td></tr>
<tr>
<td style="color:#333333;">
<b>Target</b>: 
ObservationElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">References the ObservationTarget that this feature was made on. For example 'individual1'. In a correlation matrix this could be also 'probe123'.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ObservedValue.
</td></tr>
</table>
<h3><a name="Protocol">Protocol</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements InvestigationElement</i><br>
</p>
<p>	


				The Protocol class defines parameterizable descriptions of methods;
				each protocol has a unique name within an Study. Each
				ProtocolApplication can define the ObservableFeatures it can
				observe. Also the protocol parameters can be modeled using
				ObservableFeatures (Users are expected to 'tag' the observeable
				feature by setting ObserveableFeature type as 'ProtocolParameter'.
				Examples of protocols are: SOP for blood pressure measurement used
				by UK biobank, or 'R/qtl' as protocol for statistical analysis.
				Protocol is a high level object that represents the details of
				protocols used during the investigation. The uses of Protocols to
				process BioMaterials and Data are referenced by ProtocolApplication
				(in the SDRF part of the format). Protocol has an association to
				OntologyTerm to represent the type of protocol. Protocols are
				associated with Hardware, Software and Parameters used in the
				Protocol. An example from ArrayExpress is E-MTAB-506
				<a href="ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt">
					ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/experiment/TABM/E-TABM-506/E-TABM-506.idf.txt.
				</a>
    <br/>
				The FUGE equivalent to Protocol is FuGE::Protocol.
				<br/>
				The Protocol class maps to FuGE/XGAP/MageTab Protocol, but in
				contrast to FuGE it is not required to extend protocol before use.
				The Protocol class also maps to METABASE:Form (note that components
				are solved during METABASE:Visit which can be nested). Has no
				equivalent in PaGE.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>description</b>: richtext 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Description, or reference to a description, of the protocol
</td></tr>
<tr>
<td 1>
<b>TargetFilter</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Expression that filters the InvestigationElements that can be targetted using this protocol.     This helps the user to only select from targets that matter when setting observedvalues. For example: type='individual' AND species = 'human'
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ProtocolType</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">annotation of the protocol to a well-defined ontological class.</td></tr>
<tr>
<td style="color:#333333;">
<b>Features</b>: ObservableFeature (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The features that can be observed using this protocol. For example 'length' or 'rs123534' or 'probe123'. Also protocol parameters are considered observable features as they are important to the interpretation of the observed values.</td></tr>
<tr>
<td style="color:#333333;">
<b>Contact</b>: 
Person (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">TODO Check if there can be multiple contacts.</td></tr>
<tr>
<td style="color:#333333;">
<b>subprotocols</b>: Protocol (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Subprotocols of this protocol</td></tr>
<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Protocol.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
</table>
<h3><a name="ProtocolApplication">ProtocolApplication</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements InvestigationElement</i><br>
</p>
<p>	

				A ProtocolApplication class defines the actual action of observation
				by refering to a protocol and optional ParameterValues. The name
				field can be used to label applications with a human understandeable
				tag. For example: the action of blood pressure measurement on 1000
				individuals, using a particular protocol, resulting in 1000
				associated observed values. If desired, protocols can be shared
				between Studys; in those cases one should simply refer to a protocol
				in another Study.
				<br/>
				ProtocolApplications are used in MAGE-TAB format to reference to
				protocols used, with optionally use of certain protocol parameter
				values. For example, a Source may be transformed into a Labeled
				Extract by the subsequent application of a Extraction and Labeling
				protocol. ProtocolApplication is associated with and Edge that links
				input/output, e.g. Source to Labeled Extract. The order of the
				application of protocols can be set in order to be able to
				reconstruct the left-to-right order of protocol references in
				MAGE-TAB format. The FuGE equivalent to ProtocolApplication is
				FuGE:ProtocolApplication, however input/output is modeled using
				Edge.
				<br/>
				The ProtocolApplication class maps to FuGE/XGAP ProtocolApplication,
				but in FuGE ProtocolApplications can take Material or Data (or both)
				as input and produce Material or Data (or both) as output. Similar
				to PaGE.ObservationMethod. Maps to METABASE:Visit (also note that
				METABASE:PlannedVisit allows for planning of protocol applications;
				this is outside scope for this model?).
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>time</b>: datetime 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">time when the protocol was applied.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">description field
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>protocol</b>: 
Protocol (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the protocol that is being used.</td></tr>
<tr>
<td style="color:#333333;">
<b>Performer</b>: Person (0..n)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Investigation</b>: 
Investigation (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study that this data element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>ontologyReference</b>: 
OntologyTerm (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">(Optional) Reference to the formal ontology definition for this element, e.g. 'Animal' or 'GWAS protocol'</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ProtocolApplication.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, Investigation)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within an Investigation.
</td></tr>
</table>
<h3><a name="ProtocolDocument">ProtocolDocument</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>document</b>: file 
(required)
</td>
</tr>

</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>protocol</b>: 
Protocol (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ProtocolDocument.
</td></tr>
</table>
<h3><a name="Workflow">Workflow</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Protocol</i><br>
<i> implements Identifiable, Nameable</i><br>
</p>
<p>	
A workflow is a plan to execute a series of subprotocols
				in a particular order. Each workflow elements is another protocol as
				refered to via WorkflowElement. Because Workflow extends Protocol,
				workflows can be nested just as any other protocol.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
description, 
Investigation, 
ontologyReference, 
ProtocolType, 
Features, 
TargetFilter, 
Contact, 
subprotocols, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Workflow.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an Workflow.
</td></tr>
</table>
<h3><a name="WorkflowElement">WorkflowElement</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	
Elements of a workflow are references to protocols. The
				whole workflow is a directed graph with each element pointing to the
				previousSteps that the current workflow element depends on.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Workflow</b>: 
Workflow (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Workflow this element is part of</td></tr>
<tr>
<td style="color:#333333;">
<b>protocol</b>: 
Protocol (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Protocol to be used at this workflow step</td></tr>
<tr>
<td style="color:#333333;">
<b>PreviousSteps</b>: WorkflowElement (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">Previous steps that need to be done before this protocol can be executed.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an WorkflowElement.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field name is unique within an WorkflowElement.
</td></tr>
</table>
<h3><a name="WorkflowElementParameter">WorkflowElementParameter</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	
Element parameters are the way to link workflow elements
				together. It allows override of the parameters from the previous
				step.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Value of this parameter. Can be a template of form ${other} refering to previous values in context.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>WorkflowElement</b>: 
WorkflowElement (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">To attach a parameter to a WorkflowElement</td></tr>
<tr>
<td style="color:#333333;">
<b>Parameter</b>: 
ObservableFeature (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Parameter definition.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an WorkflowElementParameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(WorkflowElement, Parameter)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Feature value is set only once per element.
</td></tr>
</table>
<h3><a name="Locus">Locus</a> (interface).</h3>
<p style="margin-top: 0px; margin-bottom: 0px;">


</p>
<p>	
 position. Typical examples of such traits are genes,
				probes and markers. Common structure for entities that have a
				genomic

</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td 1>
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td 1>
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td 1>
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td 1>
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<h3><a name="Chromosome">Chromosome</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>orderNr</b>: int 
(required)
</td>
</tr>

<tr>
<td 1>
<b>isAutosomal</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Is 'yes' when number of chromosomes is equal in male and female individuals, i.e., if not a sex chromosome.
</td></tr>
<tr>
<td 1>
<b>bpLength</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Lenght of the chromsome in base pairs.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Species</b>: 
Species (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the species this     chromosome belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Chromosome.
</td></tr>
</table>
<h3><a name="NMRBin">NMRBin</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Shift of the NMR frequency due to the chemical
				environment.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an NMRBin.
</td></tr>
</table>
<h3><a name="Clone">Clone</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 BAC clone fragment.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Clone.
</td></tr>
</table>
<h3><a name="DerivedTrait">DerivedTrait</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Any meta trait, eg. false discovery rates, P-values,
				thresholds.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DerivedTrait.
</td></tr>
</table>
<h3><a name="EnvironmentalFactor">EnvironmentalFactor</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
Experimental conditions,
			such as temperature differences, batch effects etc.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an EnvironmentalFactor.
</td></tr>
</table>
<h3><a name="Gene">Gene</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 Trait annotations specific for genes.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Main symbol this gene is known by (not necessarily unique, in constrast to 'name')
</td></tr>
<tr>
<td 1>
<b>Orientation</b>: enum 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Orientation of the gene on the genome (F=forward, R=reverse)
<br>enum_options: 'F', 'R'
</td></tr>
<tr>
<td 1>
<b>Control</b>: bool 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Indicating whether this is a 'housekeeping' gene that can be used as control.
</td></tr>
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Gene.
</td></tr>
</table>
<h3><a name="Transcript">Transcript</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for transcripts.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Gene</b>: 
Gene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene that produces this protein</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Transcript.
</td></tr>
</table>
<h3><a name="Protein">Protein</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for proteins.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>AminoSequence</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The aminoacid sequence.
</td></tr>
<tr>
<td 1>
<b>Mass</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The mass of this metabolite
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Gene</b>: 
Gene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene that produces this protein</td></tr>
<tr>
<td 1>
<b>Transcript</b>: 
Transcript (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The transcript variant that produces this protein</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Protein.
</td></tr>
</table>
<h3><a name="Metabolite">Metabolite</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for metabolites.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Formula</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The chemical formula of a metabolite.
</td></tr>
<tr>
<td 1>
<b>Mass</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The mass of this metabolite
</td></tr>
<tr>
<td 1>
<b>Structure</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The chemical structure of a metabolite (in SMILES representation).
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Metabolite.
</td></tr>
</table>
<h3><a name="Marker">Marker</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 Trait annotations specific for markers.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ReportsFor</b>: Marker (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The marker (or a subclass like 'SNP') this marker (or a subclass like 'SNP') reports for.</td></tr>
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Marker.
</td></tr>
</table>
<h3><a name="SNP">SNP</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Marker</i><br>

</p>
<p>	
A SNP is a special kind of Marker, but can also be seen as a phenotype to map against in some cases. A single-nucleotide polymorphism is a DNA sequence variation occurring when a single nucleotide in the genome (or other shared sequence) differs between members of a biological species or paired chromosomes in an individual.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
ReportsFor, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Status</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The status of this SNP, eg 'confirmed'.
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td 1>
<b>Polymorphism</b>: Polymorphism (0..n)
</td>
</tr>
<tr><td style="padding-left: 50px;">The polymorphism that belongs to this SNP.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SNP.
</td></tr>
</table>
<h3><a name="Polymorphism">Polymorphism</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
The difference of a single base discovered between two sequenced individuals.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Base</b>: enum 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The affected DNA base. Note that you can select the reference base here.
<br>enum_options: 'Reference', 'A', 'T', 'C', 'G'
</td></tr>
<tr>
<td 1>
<b>Value</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The strain/genotype for which this polymorphism was discovered. E.g. 'N2' or 'CB4856'.
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Polymorphism.
</td></tr>
</table>
<h3><a name="Probe">Probe</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 A piece of sequence that reports for the expression of
				a gene, typically spotted onto a microarray.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Mismatch</b>: bool 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Indicating whether the probe is a match
</td></tr>
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ProbeSet</b>: 
ProbeSet (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Optional: probeset this probe belongs to (e.g., in Affymetrix assays).</td></tr>
<tr>
<td style="color:#333333;">
<b>ReportsFor</b>: 
Gene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene this probe reports for.</td></tr>
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Probe.
</td></tr>
</table>
<h3><a name="Spot">Spot</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends Probe</i><br>

</p>
<p>	

				This is the spot on a microarray.
				<br/>
				Note: We don't distinquish between probes (the sequence) and spots
				(the sequence as spotted on the array).
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
Mismatch, 
ProbeSet, 
ReportsFor, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>x</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Row
</td></tr>
<tr>
<td 1>
<b>y</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Column
</td></tr>
<tr>
<td 1>
<b>GridX</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Meta Row
</td></tr>
<tr>
<td 1>
<b>GridY</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Meta Column
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Spot.
</td></tr>
<tr>
<td 1>
<b>unique(x, y, GridX, GridY)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
The combination of fields x, y, GridX, GridY is unique within an Spot.
</td></tr>
</table>
<h3><a name="ProbeSet">ProbeSet</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 A set of Probes. E.g. an Affymetrix probeset has multiple
				probes. It implements locus because sometimes you want to give the complete set of probes a range,
				for example: indicating that this set of probes spans basepair 0 through 10.000.000 on chromosome 3.
				The same information could arguably also be queried from the probes themselves, but if you have 40k probes,
				retrieving the same information from only ProbeSet (if annotated so) would be much faster.

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ProbeSet.
</td></tr>
</table>
<h3><a name="MassPeak">MassPeak</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 A peak that has been selected within a mass
				spectrometry experiment.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>MZ</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Mass over charge ratio of this peak.
</td></tr>
<tr>
<td 1>
<b>RetentionTime</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The retention-time of this peak in minutes.
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an MassPeak.
</td></tr>
</table>
<h3><a name="InvestigationFile">InvestigationFile</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends MolgenisFile</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
Extension, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Investigation</b>: 
Investigation (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the Study.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an InvestigationFile.
</td></tr>
</table>
<h3><a name="Tissue">Tissue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Tissue.
</td></tr>
</table>
<h3><a name="SampleLabel">SampleLabel</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends OntologyTerm</i><br>

</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
ontology, 
termAccession, 
definition, 
termPath, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SampleLabel.
</td></tr>
</table>
<h3><a name="Sample">Sample</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	


</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Individual</b>: 
Individual (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The individual from which this sample was taken.</td></tr>
<tr>
<td style="color:black;">
<b>Tissue</b>: 
Tissue (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The tissue from which this sample was taken.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Sample.
</td></tr>
</table>
<h3><a name="PairedSample">PairedSample</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	
 A pair of samples labeled for a two-color microarray
				experiment.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:black;">
<b>Subject1</b>: 
Individual (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The first subject</td></tr>
<tr>
<td style="color:black;">
<b>Label1</b>: 
SampleLabel (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Which channel or Fluorescent labeling is associated with the first subject</td></tr>
<tr>
<td style="color:black;">
<b>Subject2</b>: 
Individual (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The second sample</td></tr>
<tr>
<td style="color:black;">
<b>Label2</b>: 
SampleLabel (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Which channel or Fluorescent labeling is associated with the second subject</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an PairedSample.
</td></tr>
</table>
<h3><a name="Job">Job</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>OutputDataName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Name of the matrix that will be written
</td></tr>
<tr>
<td 1>
<b>Timestamp</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Datatime when the job was started.
</td></tr>
<tr>
<td 1>
<b>ComputeResource</b>: enum 
(required)
</td>
</tr>

<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Analysis</b>: 
Analysis (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Job.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(OutputDataName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
OutputDataName is unique.
</td></tr>
</table>
<h3><a name="Subjob">Subjob</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>StatusCode</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Status code of this subjob.
</td></tr>
<tr>
<td 1>
<b>StatusText</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Status text of this subjob.
</td></tr>
<tr>
<td 1>
<b>StatusProgress</b>: int 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Percentage done.
</td></tr>
<tr>
<td 1>
<b>Nr</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Number of this subjob within the job.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Job</b>: 
Job (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the job this subjob belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Subjob.
</td></tr>
</table>
<h3><a name="Analysis">Analysis</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Optional description of this type of analysis.
</td></tr>
<tr>
<td 1>
<b>TargetFunctionName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">The function used to start a specific type of analysis on the cluster.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ParameterSet</b>: 
ParameterSet (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>DataSet</b>: 
DataSet (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an Analysis.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="ParameterSet">ParameterSet</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ParameterSet.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="ParameterName">ParameterName</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Description</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Optional description of this parameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ParameterSet</b>: 
ParameterSet (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ParameterName.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, ParameterSet)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a ParameterSet.
</td></tr>
</table>
<h3><a name="ParameterValue">ParameterValue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Value</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Possible value of this parameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ParameterName</b>: 
ParameterName (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an ParameterValue.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, ParameterName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a ParameterName.
</td></tr>
</table>
<h3><a name="DataSet">DataSet</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataSet.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique.
</td></tr>
</table>
<h3><a name="DataName">DataName</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>DataSet</b>: 
DataSet (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataName.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, DataSet)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a dataset.
</td></tr>
</table>
<h3><a name="DataValue">DataValue</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Nameable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
name, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">human-readable name.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>DataName</b>: 
DataName (1..1)

</td>
</tr>

<tr>
<td style="color:#333333;">
<b>Value</b>: 
Data (1..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Possible reference of this Data.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an DataValue.
</td></tr>
<tr>
<td style="color:#333333;">
<b>unique(name, DataName)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Name is unique within a DataName.
</td></tr>
</table>
<h3><a name="SelectedParameter">SelectedParameter</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>ParameterName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied name of this parameter.
</td></tr>
<tr>
<td 1>
<b>ParameterValue</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied value of this parameter.
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Job</b>: 
Job (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SelectedParameter.
</td></tr>
</table>
<h3><a name="SelectedData">SelectedData</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">

<i> implements Identifiable</i><br>
</p>
<p>	

</p>
<p>
<u>Inherited atttributes:</u><br>
id, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>DataName</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied name of this Data
</td></tr>
<tr>
<td 1>
<b>DataValue</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Copied referenced name of this Data
</td></tr>
<tr>
<td style="color:#333333;">
<b>id</b>: int 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">automatically generated id
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>Job</b>: 
Job (1..1)

</td>
</tr>

</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an SelectedData.
</td></tr>
</table>
<h3><a name="RScript">RScript</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends InvestigationFile</i><br>

</p>
<p>	

			Proof of concept to show users can add scripts to database, to be replaced later with more generic version from compute model.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
Extension, 
Investigation, 
</p>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:black;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an RScript.
</td></tr>
</table>
<h3><a name="HemoSample">HemoSample</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservationTarget</i><br>

</p>
<p>	


</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>name</b>: string 
(required)
</td>
</tr>
<tr><td style="padding-left: 50px;">Illumina internal ID
</td></tr>
<tr>
<td 1>
<b>Platform</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Illumina Platform
</td></tr>
<tr>
<td 1>
<b>RNA_Amplification</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Method of amplification (Magnetic or Column)
</td></tr>
<tr>
<td 1>
<b>Sample</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">UMCG sample ID
</td></tr>
<tr>
<td 1>
<b>expr_CD34</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Expression of CD34, plus = expressed, minus = not expressed
</td></tr>
<tr>
<td 1>
<b>AML_normal</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Sample type
</td></tr>
<tr>
<td 1>
<b>Sample_Group</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Sample classification (ALL CD34+, AML CD34-, AML CD34+, AML MNC, BC CML, BM CD34+, BM post ASCT, BM retroviral model, CB CD34+, CB retroviral model, mob PBSC, NA, NSG, stroma)
</td></tr>
<tr>
<td 1>
<b>age</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">NA, old, young, fetal, numerical
</td></tr>
<tr>
<td 1>
<b>FAB</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">French-American-British classification system of hematologic diseases
</td></tr>
<tr>
<td 1>
<b>risk</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Risk group (Poor, intermediate, good)
</td></tr>
<tr>
<td 1>
<b>FLT3_class</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Classification based on the FLT3 gene (ITD,NA,normal,WT)
</td></tr>
<tr>
<td 1>
<b>NPM</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Classification based on NPM (cyt,NA,not determined,wt)
</td></tr>
<tr>
<td 1>
<b>PROCENT_CD34</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Percentage of CD34
</td></tr>
<tr>
<td 1>
<b>PB_BM_cb</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Sample origin (peripheral blood, bone marrow or cord blood)
</td></tr>
<tr>
<td 1>
<b>Age_years</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Age in years
</td></tr>
<tr>
<td 1>
<b>Cytogenetics</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Cytogenetics (where NK = Normal Karyotype)
</td></tr>
<tr>
<td 1>
<b>OS_months</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Overall survival
</td></tr>
<tr>
<td 1>
<b>EFS_months</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Event free survival
</td></tr>
<tr>
<td 1>
<b>CR</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Complete remission
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an HemoSample.
</td></tr>
</table>
<h3><a name="HemoGene">HemoGene</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>

</p>
<p>	
 Trait annotations specific for genes.

</p>
<p>
<u>Inherited atttributes:</u><br>
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Main symbol this gene is known by (not necessarily unique, in constrast to 'name')
</td></tr>
<tr>
<td 1>
<b>Control</b>: bool 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Indicating whether this is a 'housekeeping' gene that can be used as control.
</td></tr>
<tr>
<td 1>
<b>Orientation</b>: enum 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">Orientation of the gene on the genome (F=forward, R=reverse)
<br>enum_options: '?', 'F', 'R'
</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td 1>
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an HemoGene.
</td></tr>
</table>
<h3><a name="HemoProbe">HemoProbe</a></h3>
<p style="margin-top: 0px; margin-bottom: 0px;">
<i> extends ObservableFeature</i><br>
<i> implements Locus</i><br>
</p>
<p>	
 A piece of sequence that reports for the expression of
				a gene, typically spotted onto a microarray.
			

</p>
<p>
<u>Inherited atttributes:</u><br>
Chromosome, 
cM, 
bpStart, 
bpEnd, 
Seq, 
Symbol, 
name, 
description, 
Investigation, 
ontologyReference, 
AlternateId, 
label, 
</p>
<p>
<u>Attributes:</u>
<table>	
<tr>
<td 1>
<b>EntrezID</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The entrez ID for the probe
</td></tr>
<tr>
<td 1>
<b>RefseqID</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The refseq ID for the probe
</td></tr>
<tr>
<td style="color:#333333;">
<b>cM</b>: decimal 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">genetic map position in centi morgan (cM).
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpStart</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (5') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>bpEnd</b>: long 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">numeric basepair postion (3') on the chromosome
</td></tr>
<tr>
<td style="color:#333333;">
<b>Seq</b>: text 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">The FASTA text representation of the sequence.
</td></tr>
<tr>
<td style="color:#333333;">
<b>Symbol</b>: string 
(optional)
</td>
</tr>
<tr><td style="padding-left: 50px;">todo
</td></tr>
</table>
<p>
<u>Associations:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>ProbeSet</b>: 
ProbeSet (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Optional: probeset this probe belongs to (e.g., in Affymetrix assays).</td></tr>
<tr>
<td style="color:#333333;">
<b>ReportsFor</b>: 
HemoGene (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">The gene this probe reports for.</td></tr>
<tr>
<td style="color:#333333;">
<b>Chromosome</b>: 
Chromosome (0..1)

</td>
</tr>
<tr><td style="padding-left: 50px;">Reference to the chromosome this     position belongs to.</td></tr>
</table>
<p>
<u>Constraints:</u>
<table>	
<tr>
<td style="color:#333333;">
<b>unique(id)</b>: 
</td>
</tr>
<tr><td style="padding-left: 50px;">
Field id is unique within an HemoProbe.
</td></tr>
</table>
<br>
<br>

<h1>Supplementary figure: complete data model</h1>
<a name="__figure_of_complete_schema"></a><br>
<img src="objectmodel-uml-diagram.dot.png" style="border: solid double black;">
<br>
<a href="#_top_of_page">go to top</a>

<div align="center">Documentation generated on June 22, 2012 by MOLGENIS <a href="http://www.molgenis.org">http://www.molgenis.org</a></div>
</body></html>



